<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Evaluating gwas</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.15/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa-home"></span>
     
    Home
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://osf.io/7n4qj">
    <span class="fas fa-hand-spock-o"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Evaluating gwas</h1>

</div>


<style>
body{
  font-family: 'Oxygen', sans-serif;
  font-size: 16px;
  line-height: 24px;
}
</style>
<p>Here, we evaluate the GWAS summary statistics created in a previous step. The following properties of the summary statistics are assessed:</p>
<ul>
<li>Heritability</li>
<li>Genomic Inflation Factor</li>
<li>Univariate intercepts</li>
<li>Signal-to-noise ratio</li>
<li>Do the ldsc intercepts from the whole-brain genetic correlations correlate with phenotypic correlations?</li>
</ul>
<p><span class="math display">\[\\[0.5in]\]</span></p>
<div id="heritability" class="section level1">
<h1>Heritability</h1>
<p>Are all traits heritable?</p>
<pre class="bash"><code>#!/bin/bash -l

#SBATCH --output=~ldsc/calculate_heritabilities.out
#SBATCH --job-name=heritability_brain_volume


# specify time limit of two hours
#SBATCH --time=0-99:00

# memory per job 
#SBATCH --mem=80G

module load devtools/anaconda/2019.3-python2.7.16

~/Software/ldscore_2020_06/ldsc/ldsc.py \
--rg ~gwas_munged/Brain_stem.sumstats.gz,\
~gwas_munged/Left_accumbens_area.sumstats.gz,\
~gwas_munged/Left_amygdala.sumstats.gz,\
~gwas_munged/Left_bankssts.sumstats.gz,\
~gwas_munged/Left_caudal_anterior_cingulate.sumstats.gz,\
~gwas_munged/Left_caudal_middle_frontal.sumstats.gz,\
~gwas_munged/Left_caudate.sumstats.gz,\
~gwas_munged/Left_cuneus.sumstats.gz,\
~gwas_munged/Left_DC.sumstats.gz,\
~gwas_munged/Left_entorhinal.sumstats.gz,\
~gwas_munged/Left_frontal_pole.sumstats.gz,\
~gwas_munged/Left_fusiform.sumstats.gz,\
~gwas_munged/Left_hippocampus.sumstats.gz,\
~gwas_munged/Left_inferior_parietal.sumstats.gz,\
~gwas_munged/Left_inferior_temporal.sumstats.gz,\
~gwas_munged/Left_insula.sumstats.gz,\
~gwas_munged/Left_isthmus_cingulate.sumstats.gz,\
~gwas_munged/Left_lateral_occipital.sumstats.gz,\
~gwas_munged/Left_lateral_orbitofrontal.sumstats.gz,\
~gwas_munged/Left_lingual.sumstats.gz,\
~gwas_munged/Left_medial_orbitofrontal.sumstats.gz,\
~gwas_munged/Left_middle_temporal.sumstats.gz,\
~gwas_munged/Left_pallidum.sumstats.gz,\
~gwas_munged/Left_paracentral.sumstats.gz,\
~gwas_munged/Left_parahippocampal.sumstats.gz,\
~gwas_munged/Left_pars_opercularis.sumstats.gz,\
~gwas_munged/Left_pars_orbitalis.sumstats.gz,\
~gwas_munged/Left_pars_triangularis.sumstats.gz,\
~gwas_munged/Left_pericalcarine.sumstats.gz,\
~gwas_munged/Left_postcentral.sumstats.gz,\
~gwas_munged/Left_posterior_cingulate.sumstats.gz,\
~gwas_munged/Left_precentral.sumstats.gz,\
~gwas_munged/Left_precuneus.sumstats.gz,\
~gwas_munged/Left_putamen.sumstats.gz,\
~gwas_munged/Left_rostral_anterior_cingulate.sumstats.gz,\
~gwas_munged/Left_rostral_middle_frontal.sumstats.gz,\
~gwas_munged/Left_superior_frontal.sumstats.gz,\
~gwas_munged/Left_superior_parietal.sumstats.gz,\
~gwas_munged/Left_superior_temporal.sumstats.gz,\
~gwas_munged/Left_supramarginal.sumstats.gz,\
~gwas_munged/Left_thalamus_proper.sumstats.gz,\
~gwas_munged/Left_transverse_temporal.sumstats.gz,\
~gwas_munged/Right_accumbens_area.sumstats.gz,\
~gwas_munged/Right_amygdala.sumstats.gz,\
~gwas_munged/Right_bankssts.sumstats.gz,\
~gwas_munged/Right_caudal_anterior_cingulate.sumstats.gz,\
~gwas_munged/Right_caudal_middle_frontal.sumstats.gz,\
~gwas_munged/Right_caudate.sumstats.gz,\
~gwas_munged/Right_cuneus.sumstats.gz,\
~gwas_munged/Right_DC.sumstats.gz,\
~gwas_munged/Right_entorhinal.sumstats.gz,\
~gwas_munged/Right_frontal_pole.sumstats.gz,\
~gwas_munged/Right_fusiform.sumstats.gz,\
~gwas_munged/Right_hippocampus.sumstats.gz,\
~gwas_munged/Right_inferior_parietal.sumstats.gz,\
~gwas_munged/Right_inferior_temporal.sumstats.gz,\
~gwas_munged/Right_insula.sumstats.gz,\
~gwas_munged/Right_isthmus_cingulate.sumstats.gz,\
~gwas_munged/Right_lateral_occipital.sumstats.gz,\
~gwas_munged/Right_lateral_orbitofrontal.sumstats.gz,\
~gwas_munged/Right_lingual.sumstats.gz,\
~gwas_munged/Right_medial_orbitofrontal.sumstats.gz,\
~gwas_munged/Right_middle_temporal.sumstats.gz,\
~gwas_munged/Right_pallidum.sumstats.gz,\
~gwas_munged/Right_paracentral.sumstats.gz,\
~gwas_munged/Right_parahippocampal.sumstats.gz,\
~gwas_munged/Right_pars_opercularis.sumstats.gz,\
~gwas_munged/Right_pars_orbitalis.sumstats.gz,\
~gwas_munged/Right_pars_triangularis.sumstats.gz,\
~gwas_munged/Right_pericalcarine.sumstats.gz,\
~gwas_munged/Right_postcentral.sumstats.gz,\
~gwas_munged/Right_posterior_cingulate.sumstats.gz,\
~gwas_munged/Right_precentral.sumstats.gz,\
~gwas_munged/Right_precuneus.sumstats.gz,\
~gwas_munged/Right_putamen.sumstats.gz,\
~gwas_munged/Right_rostral_anterior_cingulate.sumstats.gz,\
~gwas_munged/Right_rostral_middle_frontal.sumstats.gz,\
~gwas_munged/Right_superior_frontal.sumstats.gz,\
~gwas_munged/Right_superior_parietal.sumstats.gz,\
~gwas_munged/Right_superior_temporal.sumstats.gz,\
~gwas_munged/Right_supramarginal.sumstats.gz,\
~gwas_munged/Right_thalamus_proper.sumstats.gz,\
~gwas_munged/Right_transverse_temporal.sumstats.gz \
--ref-ld-chr /mnt/lustre/groups/ukbiobank/Edinburgh_Data/Software/ldscore_2020_06/eur_w_ld_chr/ \
--w-ld-chr /mnt/lustre/groups/ukbiobank/Edinburgh_Data/Software/ldscore_2020_06/eur_w_ld_chr/ \
--out ~ldsc/heritability_volumes_ldsc.out

### Aim: run LDSC to get heritabilities with Bulik-Sullivan packages
### extract files in cd ~gwas_munged/ and save in a file
### using ls -d &quot;$PWD&quot;/*.gz &gt; listOfFiles.list

### Author: Anna Furtjes
### Date: 13/05/2021
### submit using: sbatch -p brc,shared ~/scripts/genetic_networks_project/ldsc/ldsc_bullik_heritability_volumes_13052021.sh

## copy onto local machine</code></pre>
<pre class="r"><code>workingd&lt;-getwd()
temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc/&quot;)
setwd(temporarywd)

heritability&lt;-read.table(&quot;heritability_brain_volumes_14052021.txt&quot;,header=T)
summary(heritability$h2_obs)</code></pre>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.0757  0.1858  0.2306  0.2345  0.2669  0.4248</code></pre>
<pre class="r"><code>library(plyr)
source(&quot;https://raw.githubusercontent.com/talgalili/R-code-snippets/master/boxplot.with.outlier.label.r&quot;)
boxplot_h2&lt;-boxplot.with.outlier.label(heritability$h2_obs,heritability$p2,xaxt=&quot;n&quot;,yaxt=&quot;n&quot;,ylab=&quot;SNP heritability&quot;,spread_text = F)
axis(side=2,las=2)</code></pre>
<p><img src="Evaluate_gwas_files/figure-html/display_h2-1.png" width="480" style="display: block; margin: auto;" /></p>
<div id="display-heritability-estimates" class="section level2">
<h2>Display heritability estimates</h2>
<pre class="r"><code>table&lt;-heritability[,c(&quot;p2&quot;,&quot;h2_obs&quot;,&quot;h2_obs_se&quot;)]
names(table)&lt;-c(&quot;Brain areas&quot;,&quot;h2 observed&quot;,&quot;SE&quot;)
table&lt;-table[order(table$`h2 observed`),]
library(DT)
datatable(table, rownames=FALSE,options = list(pageLength=5, scrollX=T))</code></pre>
<div id="htmlwidget-7cfbcec80a3bb7b647f6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7cfbcec80a3bb7b647f6">{"x":{"filter":"none","data":[["Right_frontal_pole","Left_frontal_pole","Right_caudal_anterior_cingulate","Left_caudal_anterior_cingulate","Left_entorhinal","Right_rostral_anterior_cingulate","Left_posterior_cingulate","Right_pars_triangularis","Left_pars_opercularis","Left_pars_orbitalis","Right_entorhinal","Left_pars_triangularis","Right_pars_opercularis","Right_posterior_cingulate","Left_bankssts","Left_medial_orbitofrontal","Left_transverse_temporal","Left_paracentral","Left_supramarginal","Right_paracentral","Right_bankssts","Right_pars_orbitalis","Right_transverse_temporal","Right_medial_orbitofrontal","Right_isthmus_cingulate","Left_isthmus_cingulate","Right_supramarginal","Left_inferior_temporal","Left_rostral_anterior_cingulate","Left_caudal_middle_frontal","Left_postcentral","Left_lateral_occipital","Left_fusiform","Right_superior_parietal","Left_precentral","Right_inferior_temporal","Right_parahippocampal","Left_parahippocampal","Left_superior_parietal","Right_inferior_parietal","Right_caudal_middle_frontal","Left_middle_temporal","Right_fusiform","Left_amygdala","Left_accumbens_area","Right_lateral_occipital","Right_middle_temporal","Right_postcentral","Right_rostral_middle_frontal","Right_precentral","Left_rostral_middle_frontal","Right_superior_frontal","Left_superior_temporal","Left_inferior_parietal","Right_accumbens_area","Right_amygdala","Right_cuneus","Right_lateral_orbitofrontal","Right_superior_temporal","Left_superior_frontal","Left_cuneus","Left_lingual","Right_lingual","Right_insula","Left_insula","Right_precuneus","Left_hippocampus","Left_lateral_orbitofrontal","Left_precuneus","Left_pericalcarine","Right_hippocampus","Left_thalamus_proper","Left_putamen","Left_pallidum","Right_putamen","Left_DC","Right_pericalcarine","Right_thalamus_proper","Right_DC","Right_caudate","Right_pallidum","Left_caudate","Brain_stem"],[0.0757,0.0774,0.1092,0.1212,0.1397,0.1405,0.1476,0.1493,0.1495,0.1533,0.1545,0.1618,0.1643,0.1664,0.1684,0.1725,0.1785,0.1827,0.1827,0.1831,0.1853,0.1863,0.19,0.1921,0.2086,0.2087,0.213,0.2145,0.2156,0.2173,0.2174,0.2193,0.2198,0.2252,0.2254,0.2261,0.2271,0.2291,0.2296,0.23,0.2305,0.2306,0.2311,0.2318,0.2319,0.2327,0.2382,0.2395,0.2406,0.242,0.2432,0.2483,0.2486,0.2517,0.2518,0.255,0.2569,0.259,0.262,0.2627,0.2629,0.2649,0.2689,0.2719,0.2764,0.2831,0.2867,0.2932,0.2978,0.3102,0.3211,0.3216,0.3232,0.3305,0.3306,0.3348,0.3373,0.3381,0.3482,0.3554,0.358,0.3731,0.4248],[0.0141,0.0153,0.0141,0.0151,0.0169,0.0154,0.02,0.0185,0.0158,0.017,0.0152,0.0151,0.0164,0.0173,0.0163,0.0191,0.0191,0.0188,0.0205,0.018,0.0167,0.0177,0.0182,0.0201,0.0185,0.0192,0.0196,0.019,0.0206,0.0233,0.0188,0.0214,0.0218,0.0199,0.0199,0.0183,0.0162,0.0189,0.0211,0.0204,0.0213,0.0202,0.0227,0.0211,0.022,0.0203,0.0201,0.0207,0.0209,0.0202,0.0217,0.023,0.0231,0.0189,0.022,0.0227,0.0228,0.023,0.0223,0.0245,0.0238,0.0221,0.0219,0.0253,0.0249,0.022,0.0242,0.0259,0.0216,0.0279,0.0269,0.0239,0.0286,0.0259,0.0288,0.0288,0.0275,0.0249,0.0294,0.0258,0.0277,0.0258,0.037]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Brain areas<\/th>\n      <th>h2 observed<\/th>\n      <th>SE<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><em>Note.</em> The frontal pole is only a tiny area which often is difficult to segment. This might have caused low levels of systematic variation across the GWAS resulting in very low SNP heritability captured by LDSC.</p>
<p><span class="math display">\[\\[1in]\]</span></p>
</div>
</div>
<div id="genomic-inflation-factor-for-each-gwas" class="section level1">
<h1>Genomic Inflation Factor for each GWAS</h1>
<p>Find an explanation for how to interpret the Genomic Inflation Factor <a href="https://bioinformaticsngs.wordpress.com/2016/03/08/genomic-inflation-factor-calculation/">online</a>. It is an index of confounding but it is also sensitive to true polygenic signal. It can increase with sample size, heritability and the number of causal variants (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137506/">REF</a>).</p>
<pre class="r"><code>### Calculate Genomic Inflation Factor for each of the GWAS
# Save in a list
# We will later contrast them with the univariate LDSC intercept

# load dependencies
library(data.table)
library(stringr)

# set wd to where the gwas files are stored
setwd(&quot;~/PhD/output/regenie/step2/&quot;)

# list all files of interest
files_of_interest &lt;- list.files(pattern=&quot;GWAS_22chr_noTBVcontrol_&quot;)

# create data.frame to hold Genomic Inflation Factor
table &lt;- as.data.frame(matrix(ncol = 3,nrow = length(files_of_interest)))
names(table)&lt;-c(&quot;trait&quot;,&quot;lambda&quot;,&quot;mean_chisq&quot;)

for(i in 1:length(files_of_interest)){
    # read in GWAS file
    file&lt;-fread(files_of_interest[i], header=T,data.table=F) 

    # save trait_name in table

    trait_name &lt;- str_remove(files_of_interest[i], pattern = &quot;GWAS_22chr_noTBVcontrol_&quot;)
    trait_name &lt;- str_remove(trait_name, pattern = &quot;.txt&quot;)
    table$trait[i]&lt;-trait_name

    ### Assess systematic bias in GWAS using Genomic Inflation Factor

    chisq &lt;- qchisq(1-file$P,1)
    lambda_median &lt;- median(chisq)/qchisq(0.5,1)
    table$lambda[i]&lt;-lambda_median
    
    ## calculate average association signal in all SNPs
    table$mean_chisq[i] &lt;- mean(file$CHISQ, na.rm=T)

    }


write.table(table,&quot;~/ldsc/lambda_per_volume_14052021.txt&quot;, quote=F, row.names=F, sep=&quot;\t&quot;)</code></pre>
<pre class="r"><code>temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc/&quot;)
setwd(temporarywd)

lambda&lt;-read.table(&quot;lambda_per_volume_14052021.txt&quot;,header=T)

  
# get table that holds univariate LDSC intercepts
univar_intercepts&lt;-heritability[,c(&quot;p2&quot;,&quot;h2_int&quot;,&quot;h2_int_se&quot;)]
names(univar_intercepts)&lt;-c(&quot;trait&quot;,&quot;intercept&quot;,&quot;intercept_se&quot;)
stats&lt;-merge(lambda, univar_intercepts, by = &quot;trait&quot;)


# plot lambda
x&lt;-density(stats$lambda)
plot(x, main = &quot;Distribution of lambda&quot;, xlab = &quot;Lambda (median)&quot;)</code></pre>
<p><img src="Evaluate_gwas_files/figure-html/readlambda-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>stats$round_lamba&lt;-round(stats$lambda, digits = 3)

library(DT)
datatable(stats[,c(&quot;trait&quot;,&quot;round_lamba&quot;)], rownames=FALSE, colnames = c(&quot;Traits&quot;,&quot;Genomic Inflation Factor&quot;),options = list(pageLength=5, scrollX=T))</code></pre>
<div id="htmlwidget-578232768dd7a2f5a64a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-578232768dd7a2f5a64a">{"x":{"filter":"none","data":[["Brain_stem","Left_accumbens_area","Left_amygdala","Left_bankssts","Left_caudal_anterior_cingulate","Left_caudal_middle_frontal","Left_caudate","Left_cuneus","Left_DC","Left_entorhinal","Left_frontal_pole","Left_fusiform","Left_hippocampus","Left_inferior_parietal","Left_inferior_temporal","Left_insula","Left_isthmus_cingulate","Left_lateral_occipital","Left_lateral_orbitofrontal","Left_lingual","Left_medial_orbitofrontal","Left_middle_temporal","Left_pallidum","Left_paracentral","Left_parahippocampal","Left_pars_opercularis","Left_pars_orbitalis","Left_pars_triangularis","Left_pericalcarine","Left_postcentral","Left_posterior_cingulate","Left_precentral","Left_precuneus","Left_putamen","Left_rostral_anterior_cingulate","Left_rostral_middle_frontal","Left_superior_frontal","Left_superior_parietal","Left_superior_temporal","Left_supramarginal","Left_thalamus_proper","Left_transverse_temporal","Right_accumbens_area","Right_amygdala","Right_bankssts","Right_caudal_anterior_cingulate","Right_caudal_middle_frontal","Right_caudate","Right_cuneus","Right_DC","Right_entorhinal","Right_frontal_pole","Right_fusiform","Right_hippocampus","Right_inferior_parietal","Right_inferior_temporal","Right_insula","Right_isthmus_cingulate","Right_lateral_occipital","Right_lateral_orbitofrontal","Right_lingual","Right_medial_orbitofrontal","Right_middle_temporal","Right_pallidum","Right_paracentral","Right_parahippocampal","Right_pars_opercularis","Right_pars_orbitalis","Right_pars_triangularis","Right_pericalcarine","Right_postcentral","Right_posterior_cingulate","Right_precentral","Right_precuneus","Right_putamen","Right_rostral_anterior_cingulate","Right_rostral_middle_frontal","Right_superior_frontal","Right_superior_parietal","Right_superior_temporal","Right_supramarginal","Right_thalamus_proper","Right_transverse_temporal"],[1.177,1.097,1.108,1.088,1.062,1.116,1.164,1.122,1.138,1.082,1.047,1.118,1.124,1.124,1.109,1.137,1.105,1.116,1.144,1.119,1.085,1.124,1.151,1.102,1.111,1.089,1.087,1.102,1.132,1.116,1.082,1.119,1.153,1.149,1.092,1.128,1.126,1.124,1.129,1.107,1.144,1.091,1.134,1.122,1.1,1.066,1.125,1.164,1.134,1.147,1.092,1.052,1.127,1.134,1.114,1.112,1.125,1.108,1.127,1.142,1.125,1.109,1.122,1.156,1.101,1.111,1.082,1.094,1.077,1.15,1.129,1.09,1.122,1.137,1.143,1.076,1.123,1.123,1.119,1.131,1.111,1.149,1.084]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Traits<\/th>\n      <th>Genomic Inflation Factor<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><span class="math display">\[\\[1in]\]</span></p>
</div>
<div id="univariate-intercepts" class="section level1">
<h1>Univariate intercepts</h1>
<p>The univariate LDSC intercept is an index of stratification confounding. It gives the expected chi-squared value for SNPs with no effect, as, under the null hypothesis, the expected mean of the chi-squared statistic is itself (i.e. 1.0).</p>
<p>Univariate LDSC intercepts close to 1 indicate that the GWAS calculation appropriately controlled for population stratification. When the LDSC intercept is much larger than 1.0, stratification is indicated because the expected value for SNPs with no effect under the null hypothesis is still association signal.</p>
<pre class="r"><code># plot intercepts
x&lt;-density(stats$intercept)
plot(x, main = &quot;Distribution of intercepts&quot;, xlab = &quot;Intercepts&quot;)</code></pre>
<p><img src="Evaluate_gwas_files/figure-html/univariate_intercepts-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># also we are interested to know whether the confidence intervals around the intercept all overlap with 1
stats$uci&lt;-round(stats$intercept+(1.96*stats$intercept_se), digits = 2)
stats$lci&lt;-round(stats$intercept-(1.96*stats$intercept_se), digits = 3)
stats$CI95 &lt;- paste(stats$lci,&quot; - &quot;, stats$uci)
stats$round_intercept&lt;-round(stats$intercept, digits = 3)

datatable(stats[,c(&quot;trait&quot;,&quot;round_intercept&quot;,&quot;CI95&quot;)], rownames=FALSE, colnames = c(&quot;Traits&quot;,&quot;Intercept&quot;,&quot;95% Confidence Interval&quot;),options = list(pageLength=5, scrollX=T))</code></pre>
<div id="htmlwidget-54ee5ed65c4d20f0d15f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-54ee5ed65c4d20f0d15f">{"x":{"filter":"none","data":[["Brain_stem","Left_accumbens_area","Left_amygdala","Left_bankssts","Left_caudal_anterior_cingulate","Left_caudal_middle_frontal","Left_caudate","Left_cuneus","Left_DC","Left_entorhinal","Left_frontal_pole","Left_fusiform","Left_hippocampus","Left_inferior_parietal","Left_inferior_temporal","Left_insula","Left_isthmus_cingulate","Left_lateral_occipital","Left_lateral_orbitofrontal","Left_lingual","Left_medial_orbitofrontal","Left_middle_temporal","Left_pallidum","Left_paracentral","Left_parahippocampal","Left_pars_opercularis","Left_pars_orbitalis","Left_pars_triangularis","Left_pericalcarine","Left_postcentral","Left_posterior_cingulate","Left_precentral","Left_precuneus","Left_putamen","Left_rostral_anterior_cingulate","Left_rostral_middle_frontal","Left_superior_frontal","Left_superior_parietal","Left_superior_temporal","Left_supramarginal","Left_thalamus_proper","Left_transverse_temporal","Right_accumbens_area","Right_amygdala","Right_bankssts","Right_caudal_anterior_cingulate","Right_caudal_middle_frontal","Right_caudate","Right_cuneus","Right_DC","Right_entorhinal","Right_frontal_pole","Right_fusiform","Right_hippocampus","Right_inferior_parietal","Right_inferior_temporal","Right_insula","Right_isthmus_cingulate","Right_lateral_occipital","Right_lateral_orbitofrontal","Right_lingual","Right_medial_orbitofrontal","Right_middle_temporal","Right_pallidum","Right_paracentral","Right_parahippocampal","Right_pars_opercularis","Right_pars_orbitalis","Right_pars_triangularis","Right_pericalcarine","Right_postcentral","Right_posterior_cingulate","Right_precentral","Right_precuneus","Right_putamen","Right_rostral_anterior_cingulate","Right_rostral_middle_frontal","Right_superior_frontal","Right_superior_parietal","Right_superior_temporal","Right_supramarginal","Right_thalamus_proper","Right_transverse_temporal"],[1.011,1.002,1.019,1.005,0.997,1.017,1.008,1.026,1.004,1.015,1.014,1.008,1.017,1.006,1.004,1.026,1.012,1.03,1.022,1.015,1.014,1.023,1.006,1.017,1.004,1.015,1.021,1.026,1.025,1.017,1.016,1.021,1.016,1.017,1,1.027,1.018,1.029,1.017,1.024,1.01,1.024,1.026,1.015,1.005,1.013,1.008,1.005,1.035,1.003,1.017,1.018,1.016,1.011,1.012,1.014,1.015,1.016,1.02,1.028,1.016,1.024,1.02,1.008,1.013,1.009,1.005,1.017,1.015,1.022,1.017,1.009,1.014,1.014,1.014,1.003,1.022,1.017,1.019,1.022,1.019,1.014,1.02],["0.99  -  1.03","0.987  -  1.02","1.002  -  1.04","0.992  -  1.02","0.984  -  1.01","0.999  -  1.03","0.991  -  1.03","1.009  -  1.04","0.986  -  1.02","1  -  1.03","1.001  -  1.03","0.991  -  1.02","0.998  -  1.03","0.99  -  1.02","0.989  -  1.02","1.008  -  1.04","0.996  -  1.03","1.013  -  1.05","1.004  -  1.04","1  -  1.03","0.999  -  1.03","1.007  -  1.04","0.99  -  1.02","1.002  -  1.03","0.989  -  1.02","1.001  -  1.03","1.008  -  1.03","1.012  -  1.04","1.007  -  1.04","1.002  -  1.03","1.001  -  1.03","1.003  -  1.04","1  -  1.03","0.999  -  1.03","0.985  -  1.02","1.01  -  1.04","1  -  1.04","1.013  -  1.04","0.998  -  1.04","1.009  -  1.04","0.993  -  1.03","1.009  -  1.04","1.008  -  1.04","0.999  -  1.03","0.991  -  1.02","1  -  1.03","0.993  -  1.02","0.988  -  1.02","1.019  -  1.05","0.984  -  1.02","1.003  -  1.03","1.005  -  1.03","1  -  1.03","0.993  -  1.03","0.996  -  1.03","0.998  -  1.03","0.997  -  1.03","1.002  -  1.03","1.005  -  1.03","1.011  -  1.04","1.001  -  1.03","1.009  -  1.04","1.004  -  1.03","0.991  -  1.02","0.998  -  1.03","0.994  -  1.02","0.991  -  1.02","1.003  -  1.03","0.998  -  1.03","1.006  -  1.04","1  -  1.03","0.994  -  1.02","0.998  -  1.03","0.998  -  1.03","0.995  -  1.03","0.989  -  1.02","1.005  -  1.04","1  -  1.03","1.003  -  1.03","1.007  -  1.04","1.005  -  1.03","0.997  -  1.03","1.005  -  1.03"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Traits<\/th>\n      <th>Intercept<\/th>\n      <th>95% Confidence Interval<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><span class="math display">\[\\[1in]\]</span></p>
</div>
<div id="signal-to-noise-ratio" class="section level1">
<h1>Signal-to-noise ratio</h1>
<p>To get an index for the ratio of true signal to spurious signal, we can divide the mean chi-squared value within a set of summary statistics by the univariate intercept-1.</p>
<p>This mean chi-squared value is an index of the average association signal across all SNPs. However, it includes the chi-squared values expected under the null hypothesis, which is 1.0 (see above under <em>Univariate intercepts</em>). In order to get the mean signal beyond the null hypothesis, we subtract 1.0 from the mean chi-squared values across all SNPs. Finally, the ratio between mean chi-squared - 1 and intercept - 1 is the ratio of true signal to confounding. A comprehensive and intuitive explanation of the interpretation of this ratio can be found <a href="https://www.nature.com/articles/s41588-018-0144-6#Sec3">here</a>(Section 4 in Supplementary materials). Note to self: Keep in mind, that there is no point considering confidence intervals in this case because the ldsc heritability is a signal-to-noise ratio and already has a standard error.</p>
<pre class="r"><code>#calculate the ratio of the mean chisq-1 to the univariate intercept-1 to get a sense of the ratio of true signal to spurious signal
#Mean chisq -1 / Intercept-1

# first we round one intercept to the second digit because we have one trait where the intercept is slightly below 1 and results in a negative ratio
# note that this value will result in an infite value instead
trait_intercept_below1&lt;-stats[which(stats$intercept&lt;1),&quot;trait&quot;]
intercept_below1&lt;-stats[which(stats$intercept&lt;1),&quot;intercept&quot;]
stats$intercept[which(stats$intercept&lt;1)]&lt;-round(stats$intercept[which(stats$intercept&lt;1)],digits = 2)

stats$ratio&lt;-(stats$mean_chisq - 1)/(stats$intercept-1)


# display as table
stats$ratio&lt;-round(stats$ratio, digits = 3)
stats$mean_chisq&lt;-round(stats$mean_chisq, digits = 3)
# the signal to noise ratio for one trait is Inf, but is displayed as an empty cell
stats[which(stats$ratio&gt;10000),&quot;ratio&quot;]&lt;-&quot;Inf&quot;

datatable(stats[,c(&quot;trait&quot;,&quot;ratio&quot;,&quot;mean_chisq&quot;,&quot;intercept&quot;)], rownames=FALSE, colnames = c(&quot;Traits&quot;,&quot;Signal to noise ratio&quot;, &quot;Mean chisq&quot;,&quot;Univariate intercept&quot;),options = list(pageLength=5, scrollX=T))</code></pre>
<div id="htmlwidget-3159cad579b2a58bcc92" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3159cad579b2a58bcc92">{"x":{"filter":"none","data":[["Brain_stem","Left_accumbens_area","Left_amygdala","Left_bankssts","Left_caudal_anterior_cingulate","Left_caudal_middle_frontal","Left_caudate","Left_cuneus","Left_DC","Left_entorhinal","Left_frontal_pole","Left_fusiform","Left_hippocampus","Left_inferior_parietal","Left_inferior_temporal","Left_insula","Left_isthmus_cingulate","Left_lateral_occipital","Left_lateral_orbitofrontal","Left_lingual","Left_medial_orbitofrontal","Left_middle_temporal","Left_pallidum","Left_paracentral","Left_parahippocampal","Left_pars_opercularis","Left_pars_orbitalis","Left_pars_triangularis","Left_pericalcarine","Left_postcentral","Left_posterior_cingulate","Left_precentral","Left_precuneus","Left_putamen","Left_rostral_anterior_cingulate","Left_rostral_middle_frontal","Left_superior_frontal","Left_superior_parietal","Left_superior_temporal","Left_supramarginal","Left_thalamus_proper","Left_transverse_temporal","Right_accumbens_area","Right_amygdala","Right_bankssts","Right_caudal_anterior_cingulate","Right_caudal_middle_frontal","Right_caudate","Right_cuneus","Right_DC","Right_entorhinal","Right_frontal_pole","Right_fusiform","Right_hippocampus","Right_inferior_parietal","Right_inferior_temporal","Right_insula","Right_isthmus_cingulate","Right_lateral_occipital","Right_lateral_orbitofrontal","Right_lingual","Right_medial_orbitofrontal","Right_middle_temporal","Right_pallidum","Right_paracentral","Right_parahippocampal","Right_pars_opercularis","Right_pars_orbitalis","Right_pars_triangularis","Right_pericalcarine","Right_postcentral","Right_posterior_cingulate","Right_precentral","Right_precuneus","Right_putamen","Right_rostral_anterior_cingulate","Right_rostral_middle_frontal","Right_superior_frontal","Right_superior_parietal","Right_superior_temporal","Right_supramarginal","Right_thalamus_proper","Right_transverse_temporal"],["28.843","98.176","7.473","23.033","Inf","9.351","26.607","6.687","56.749","6.406","3.86","25.221","11.292","28.36","30.171","6.912","11.162","5.602","9.511","12.259","8.79","6.449","41.732","7.767","39.089","7.445","5.018","4.695","7.891","9.139","6.619","7.66","12.659","11.607","1314.754","6.577","10.686","5.846","10.804","5.974","22.366","5.534","6.627","10.743","22.783","6.267","18.74","40.217","5.503","86.67","6.122","3.414","12.831","19.57","13.837","10.927","12.144","8.511","9.07","6.2","12.869","6.023","7.856","32.113","10.352","15.274","22.974","7.038","7.339","9.897","10.429","12.976","11.4","14.015","14.761","28.927","7.951","10.107","8.453","8.609","8.418","16.384","6.233"],[1.314,1.167,1.142,1.117,1.07,1.155,1.226,1.171,1.255,1.094,1.054,1.194,1.187,1.173,1.136,1.181,1.131,1.17,1.204,1.188,1.123,1.15,1.234,1.134,1.137,1.111,1.106,1.121,1.197,1.155,1.105,1.159,1.206,1.196,1.131,1.175,1.189,1.168,1.187,1.146,1.221,1.131,1.17,1.163,1.123,1.083,1.154,1.213,1.191,1.269,1.105,1.062,1.199,1.215,1.169,1.148,1.18,1.137,1.18,1.171,1.201,1.146,1.155,1.254,1.13,1.139,1.113,1.118,1.109,1.22,1.178,1.121,1.163,1.203,1.199,1.095,1.171,1.171,1.159,1.192,1.161,1.229,1.122],[1.0109,1.0017,1.019,1.0051,1,1.0166,1.0085,1.0255,1.0045,1.0147,1.0141,1.0077,1.0166,1.0061,1.0045,1.0262,1.0117,1.0303,1.0215,1.0153,1.014,1.0232,1.0056,1.0173,1.0035,1.0149,1.0212,1.0257,1.025,1.017,1.0158,1.0207,1.0163,1.0169,1.0001,1.0266,1.0177,1.0287,1.0173,1.0244,1.0099,1.0236,1.0257,1.0152,1.0054,1.0132,1.0082,1.0053,1.0347,1.0031,1.0172,1.0181,1.0155,1.011,1.0122,1.0135,1.0148,1.0161,1.0198,1.0276,1.0156,1.0243,1.0197,1.0079,1.0126,1.0091,1.0049,1.0167,1.0148,1.0222,1.0171,1.0093,1.0143,1.0145,1.0135,1.0033,1.0215,1.0169,1.0188,1.0223,1.0191,1.014,1.0196]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Traits<\/th>\n      <th>Signal to noise ratio<\/th>\n      <th>Mean chisq<\/th>\n      <th>Univariate intercept<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[2,3]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p>Note that one trait (Left_caudal_anterior_cingulate) has a univariate intercept below, but very close to 1 (intercept = 0.9971).</p>
<p><span class="math display">\[\\[1in]\]</span></p>
<div id="plot-ldsc-intercepts-for-genetic-correlations-across-the-brain" class="section level2">
<h2>Plot LDSC intercepts for genetic correlations across the brain</h2>
<pre class="r"><code>### plot intercepts of genetic correlation matrices
rm(list=ls())
## this script is displaying the heatmaps of the networks

# load dependencies
library(stringr)
library(reshape2)
library(ggplot2)

# load data and name it according to network
workingd&lt;-getwd()
temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc/&quot;)
setwd(temporarywd)

networks&lt;-list.files(pattern=&quot;.RData&quot;)
network_names&lt;-str_replace(networks,pattern=&quot;.RData&quot;,replacement = &quot;&quot;)

for(i in 1:length(networks)){
load(networks[i])
  name&lt;-network_names[i]
  assign(name,LDSCoutput)
}

## name all columns in $I after brain regions and round $I
for(i in network_names){
  output&lt;-get(i)
  dimnames(output$I)[[1]]&lt;-dimnames(output$S)[[2]]
  dimnames(output$I)[[2]]&lt;-dimnames(output$S)[[2]]
  name&lt;-i
  assign(name,output)
  output$I&lt;-round(output$I,digits = 2)
  name_cor&lt;-paste0(&quot;intercepts_&quot;,i)
  assign(name_cor,output$I)
}

# count number of correlation matrices
#length(ls(pattern=&quot;intercepts_&quot;))

# create vector containing names of the correlation matrices
matrices&lt;-ls(pattern=&quot;intercepts_&quot;)

## plot the rounded correlation matrices

# get lower triangle of matrix
for(i in matrices){
  cormatrix&lt;- get(i)
  get_lower_tri&lt;-function(cormatrix){
    cormatrix[upper.tri(cormatrix)] &lt;- NA
    return(cormatrix)
  }
  lower_triangle&lt;-get_lower_tri(cormatrix)
  lower_triangle&lt;-reshape2::melt(lower_triangle)
  lower_triangle$corrected&lt;-with(lower_triangle, ifelse(value&gt;=1.00,1.00,value))

  #print(i)
  #print(lower_triangle)
  
  heatmap&lt;-
    ggplot(data=lower_triangle, aes(Var1,Var2,fill=corrected))+
    geom_tile(color = &quot;white&quot;)+
    theme_minimal()+ 
    theme_bw()+
    theme(axis.text.x = element_text(angle = 45, vjust = 1, 
                                     size = 8, hjust = 1))+
    theme(axis.text.x = element_text(angle = 45, vjust = 1, 
                                     size = 8, hjust = 1))+
    theme(axis.title.x=element_blank(),
          axis.title.y = element_blank(),
          panel.grid.major = element_blank(),
          panel.border = element_blank(),
          axis.ticks = element_blank(),
          legend.justification = c(1,0),
          #legend.position = c(0.45,0.7),
          legend.direction = &quot;horizontal&quot;)+  
    scale_fill_gradient2(low=&quot;gold4&quot;,high=&quot;darkorange3&quot;,mid =&quot;gold&quot;,
                         midpoint=0,limit=c(-1,1),na.value=&quot;white&quot;,
                         name=&quot;LDSC intercepts&quot;)+
    guides(fill=guide_colorbar(barwidth = 7,barheight = 1,title.position = &quot;top&quot;,title.hjust = 0.5))+
    coord_fixed()
  
  heatmap&lt;-heatmap+
    geom_text(aes(Var1,Var2,label=value),color=&quot;black&quot;,size=2)
  
  name&lt;-paste0(i,&quot;_heatmap&quot;)
  assign(name,heatmap)
}</code></pre>
<div class="figure" style="text-align: center">
<img src="Evaluate_gwas_files/figure-html/intercepts1-1.png" alt="Central Executive" width="80%" />
<p class="caption">
Central Executive
</p>
</div>
<div class="figure" style="text-align: center">
<img src="Evaluate_gwas_files/figure-html/intercepts1-2.png" alt="Cingulo-opercular" width="80%" />
<p class="caption">
Cingulo-opercular
</p>
</div>
<div class="figure" style="text-align: center">
<img src="Evaluate_gwas_files/figure-html/intercepts2-1.png" alt="Default Mode" width="100%" />
<p class="caption">
Default Mode
</p>
</div>
</div>
<div id="ldsc-intercepts-across-the-whole-brain" class="section level2">
<h2>LDSC intercepts across the whole brain</h2>
<p><img src="Evaluate_gwas_files/figure-html/whole_brain_intercepts-1.png" width="672" style="display: block; margin: auto;" /></p>
<hr />
</div>
<div id="do-the-ldsc-intercepts-correlate-with-the-phenotypic-correlations-between-brain-areas" class="section level2">
<h2>Do the LDSC intercepts correlate with the phenotypic correlations between brain areas?</h2>
<p>The intercept depends on the phenotypic correlation between traits which can be negative or positive. Thus, we plot the intercepts as a function of the pairwise phenotypic correlation between traits. Based on LDSC theory, it should look like a straight line + small variation due to sampling error <a href="https://doi.org/10.1101/310565">REF</a>.</p>
<pre class="r"><code># load intercepts
workingd&lt;-getwd()
temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc/&quot;)
setwd(temporarywd)

load(&quot;whole_brain.RData&quot;)
ldscoutput&lt;-LDSCoutput_wholebrain
# name rows and columns
dimnames(ldscoutput$I)[[1]]&lt;-dimnames(ldscoutput$S)[[2]]
dimnames(ldscoutput$I)[[2]]&lt;-dimnames(ldscoutput$S)[[2]]

# keep lower matrix
cormatrix&lt;-ldscoutput$I
get_lower_tri&lt;-function(cormatrix){
  cormatrix[upper.tri(cormatrix)] &lt;- NA
  return(cormatrix)
}
intercepts&lt;-get_lower_tri(cormatrix)
diag(intercepts)&lt;-NA
intercepts&lt;-reshape2::melt(intercepts)
names(intercepts)&lt;-c(&quot;region1&quot;,&quot;region2&quot;,&quot;intercepts&quot;)

# load phenotypic correlations
temporarywd_pheno&lt;-paste0(workingd,&quot;/data_my_own/Pheno_preparation/&quot;)
setwd(temporarywd_pheno)

load(&quot;pheno_decomposition.RData&quot;)

phenotypic_cor&lt;-cor_matrix

# sort same as intercept matrix
names&lt;-dimnames(ldscoutput$S)[[2]]
phenotypic_cor&lt;-phenotypic_cor[names,names]

# get lower triangle
phenotypic_cor&lt;-get_lower_tri(phenotypic_cor)
phenotypic_cor&lt;-reshape2::melt(phenotypic_cor)
names(phenotypic_cor)&lt;-c(&quot;region1&quot;,&quot;region2&quot;,&quot;phenotypic_cor&quot;)

intercept_pheno&lt;-merge(intercepts,phenotypic_cor,by=c(&quot;region1&quot;,&quot;region2&quot;))

plot(intercept_pheno$phenotypic_cor,intercept_pheno$intercepts,xlab=&quot;Phenotypic correlations&quot;,ylab=&quot;LDSC intercepts&quot;)
abline(lm(intercept_pheno$intercepts~intercept_pheno$phenotypic_cor,na.action = &quot;na.omit&quot;),col=&quot;red&quot;)</code></pre>
<p><img src="Evaluate_gwas_files/figure-html/intercepts_phenotypic_cor-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(intercept_pheno$intercepts~intercept_pheno$phenotypic_cor))</code></pre>
<pre><code>## 
## Call:
## lm(formula = intercept_pheno$intercepts ~ intercept_pheno$phenotypic_cor)
## 
## Residuals:
##        Min         1Q     Median         3Q        Max 
## -0.0247109 -0.0041577 -0.0001201  0.0039923  0.0251699 
## 
## Coefficients:
##                                  Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)                    -0.0037274  0.0002879  -12.95   &lt;2e-16 ***
## intercept_pheno$phenotypic_cor  0.9798414  0.0009259 1058.22   &lt;2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.00611 on 3401 degrees of freedom
##   (3486 observations deleted due to missingness)
## Multiple R-squared:  0.997,  Adjusted R-squared:  0.997 
## F-statistic: 1.12e+06 on 1 and 3401 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code># how many intercepts are there? Should be 3403
#sum(!is.na(intercepts$intercepts))</code></pre>
<hr />
</div>
</div>

&nbsp;
<hr />
<a href = "https://www.kcl.ac.uk/">
<p style="text-align: center"><img src="KCL_logo.jpg" href="https://www.kcl.ac.uk/" style="width:70px;height:50px"> 
</a>
</p>

<a href = "https://www.kcl.ac.uk/people/anna-furtjes">
<p style="text-align: center;">By Anna Elisabeth Fürtjes</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>anna.furtjes@kcl.ac.uk</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
    <a href="https://twitter.com/Anna_Furtjes" class="fa fa-twitter"></a>
</p>

&nbsp;


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
