---
title: "GWAS calculation"
highlighter: prettify
output: 
  html_document:
    code_folding: show
    toc: yes
    toc_depth: 2
    toc_float: yes
    theme: readable
    includes:
      after_body: footer.html   
---
<style>
body{
  font-family: 'Oxygen', sans-serif;
  font-size: 16px;
  line-height: 24px;
}
</style>



```{r setup, include=F,echo=F}
require(knitr)
opts_knit$set(root.dir = "C:/Users/k1894405/OneDrive - King's College London/PhD/Projects/Genomic SEM project/")
knitr::opts_chunk$set(echo = TRUE, fig.align="center")
```

Here, we calculate the genome-wide association study summary statistics for each of our 83 brain volumes. 
The input data used was generated with the code displayed here:

* [Processing phenotypic data](Processing_phenotypic_data.html)

* [Genetic quality control](Genetic_qc.html)

We used the [regenie software](https://rgcgithub.github.io/regenie/options/) for the calculation.


$$\\[0.5in]$$


## Regenie Step 1

```{bash step1, echo=T, eval=F}
#!/bin/bash -l

#SBATCH --mem=16G
#SBATCH --nodes=1
#SBATCH --ntasks=16
#SBATCH --partition brc,shared
#SBATCH --output=~/anna/PhD/output/regenie/step2regenie.out
#SBATCH --job-name=regenie



# set input paths
cleandatapathpheno=~/anna/PhD/output/pheno_preparation/
cleandatapathgeno=~/anna/PhD/output/geno_qc/
ukbpath=/scratch/datasets/ukbiobank/ukb18177/


# set output paths
outputpath=~/anna/PhD/output/regenie/
scriptspath=~/anna/PhD/scripts/regenie/

cd  ${outputpath}

# regenie doesn't like bim files with chrom nums greater than 24
# also I don't need the X chromosome

module load apps/plink/1.9.0b6.10

plink \
--bed ${ukbpath}genotyped/ukb_binary_v2.bed  \
--bim ${ukbpath}genotyped/ukb_binary_v2.bim \
--fam ${ukbpath}genotyped/wukb18177_cal_chr1_v2_s488264.fam \
--chr 1-22 \
--make-bed \
--out ${outputpath}tmp_ukb18177_glanville_binary_pre_qc_no26X

# load regenie module
module load utilities/use.dev
module load apps/regenie/2.0.1-mkl

regenie \
  --step 1 \
  --bed ${outputpath}tmp_ukb18177_glanville_binary_pre_qc_no26X \
  --extract ${cleandatapathgeno}geneticQC_UKB_15042021__MAF0.01_GENO0.02_MIND0.02_CAUC1_UKBQC1_UNREL0.044_HWE0.00000001_SEX1.snplist \
  --phenoFile ${cleandatapathpheno}target_phenotypes_83volumes_gwas.txt \
  --keep ${cleandatapathgeno}geneticQC_UKB_15042021__MAF0.01_GENO0.02_MIND0.02_CAUC1_UKBQC1_UNREL0.044_HWE0.00000001_SEX1.fam \
  --catCovarList sex,batch \
  --maxCatLevels 106 \
  --covarFile ${cleandatapathpheno}covariates_volume_gwas.txt \
  --bsize 1000 \
  --lowmem \
  --lowmem-prefix ${outputpath}tmp_rg \
  --out ${outputpath}fit_out


### This script is an adapted version of Ryan Arathimos' script at https://github.com/applyfun/gwas_methods/blob/main/regenie_gwas_step1.sh

### sbatch -p shared,brc ~/anna/PhD/scripts/regenie/regenie_gwas_step1.sh
```

Note that the plink step prior to step 1 in regenie resulted in another 16,000 SNPs to be removed. Therefore, the final GWAS analysis included 571,257 out of 587,583 directly genotyped SNPs.

<br/>
<br/>

## Regenie Step 2

```{bash step2, echo=T, eval=F}
#!/bin/sh -l

#SBATCH --job-name=regenie_step2
#SBATCH --partition brc,shared
#SBATCH --output=~/anna/PhD/output/regenie/step2/array_regenie.%A_%a.out
#SBATCH --cpus-per-task=8
#SBATCH --mem-per-cpu=8G 
#SBATCH --array=1-22%22


cd ~/anna/PhD/scripts/regenie/

# set input paths
cleandatapathpheno=~/anna/PhD/output/pheno_preparation/
cleandatapathgeno=~/anna/PhD/output/geno_qc/
imputeddata=/mnt/lustre/groups/ukbiobank/ukb18177_glanville/imputed/

# set output paths
outputpath=~/anna/PhD/output/regenie/
scriptspath=~/anna/PhD/scripts/regenie/

# load regenie module and assign array task ID as Chromosome number
echo ${SLURM_ARRAY_TASK_ID}
CHR=${SLURM_ARRAY_TASK_ID}
echo ${CHR} 

module load utilities/use.dev
module load apps/regenie/2.0.1-mkl

regenie \
  --step 2 \
  --bgen ${imputeddata}ukb18177_glanville_imp_chr${CHR}_MAF1_INFO4_v1.bgen \
  --sample ${imputeddata}ukb18177_glanville_chr1.sample \
  --covarFile ${cleandatapathpheno}covariates_volume_gwas.txt \
  --catCovarList sex,batch \
  --maxCatLevels 106 \
  --phenoFile ${cleandatapathpheno}target_phenotypes_83volumes_gwas.txt \
  --keep ${cleandatapathgeno}geneticQC_UKB_15042021__MAF0.01_GENO0.02_MIND0.02_CAUC1_UKBQC1_UNREL0.044_HWE0.00000001_SEX1.fam \
  --bsize 400 \
  --minINFO 0.4 \
  --minMAC 5 \
  --pred ${outputpath}fit_out_pred.list \
  --split \
  --out ${outputpath}step2/regenie_chr_${CHR}_out

echo Done

### This script is an adapted version of Ryan Arathimos' script at https://github.com/applyfun/gwas_methods/blob/main/regenie_gwas_step2.sh

### sbatch -p brc,shared ~/anna/PhD/scripts/regenie/regenie_gwas_step2.sh
```

