<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Hypothesis 3</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.15/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://osf.io/7n4qj">
    <span class="fa fa-hand-spock-o"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Hypothesis 3</h1>

</div>


<style>
body{
  font-family: 'Oxygen', sans-serif;
  font-size: 16px;
  line-height: 24px;
}
</style>
<p>Using genomic structural equation modelling (genomic SEM; Grotzinger et al., 2019), we will extract a latent genetic g factor which represents common genetic variation shared by seven cognitive traits (de la Fuente et al., 2019) and correlate it with brainwide and network-specific genetic indices. As brain volumes and intelligence genetically overlap, we hypothesise that the genetic whole-brain scores positively correlate with genetic g. Additionally, we hypothesise, on the basis of previous phenotypic findings, that the central executive network shows a stronger genetic correlation than other networks, when accounting for the number of included nodes.</p>
<p><br> <br></p>
<div id="calculate-pc-sumstats" class="section level1">
<h1>Calculate PC sumstats</h1>
<p><img src="methods_visualisation.png" /></p>
<p><br> <br></p>
<hr />
<div id="format-gwas-as-expected-by-gwama-software" class="section level2">
<h2>1. Format GWAS as expected by GWAMA software</h2>
<p>The following code relates to step 4 in the illustration above.</p>
<pre class="r"><code>##### format GWAS files for GWAMA
# documentation of the Baselman function
# https://github.com/baselmans/multivariate_GWAMA/blob/master/README.md

# required format (same column names, same order):
#  SNPID = RS number
#  CHR = chromosome
#  BP = base pair
#  EA = effect allele
#  OA = other allele
#  EAF = frequency effect allele
#  N = sample size
#  Z = z-score
#  P = p-value
#___________________________________________________________________________________________________________________________________

## this script will only consider Hapmap3 SNPs in already munged files 


#load dependencies
library(data.table)
library(stringr)
library(R.utils)


## read in one original gwas for extra info on BP, CHR, P
# both OAFREQ and SE should not change even when flipping the effects when A1 and EA don&#39;t match
setwd(&quot;~/output/regenie/step2/&quot;)

ref_files&lt;-list.files(pattern=&quot;GWAS_22chr_noTBVcontrol_&quot;)

if(length(ref_files)!=83){print(&quot;Detected the wrong number of reference files&quot;);break}else{print(&quot;As expected, we have 83 reference files&quot;)}


# read in munged gwas files that were used in the PC analyses
setwd(&quot;~/output/genetic_networks_project/gwas_munged/&quot;)

munged_files&lt;-list.files(pattern=&quot;.sumstats.gz&quot;)
if(length(munged_files)!=83){print(&quot;Detected the wrong number of munged files&quot;);break}else{print(&quot;As expected, we have 83 munged files&quot;)}

all_files&lt;-data.frame(munged_files, ref_files)
print(all_files)


for (i in 1:nrow(all_files)){
    print(paste0(&quot;Now processing &quot;,i,&quot;st/nd/th trait&quot;))

    ### test if matched munged and ref files are for same phenotype
    ref_trait&lt;-str_remove(all_files$ref_files[i], pattern=&quot;GWAS_22chr_noTBVcontrol_&quot;)
    ref_trait&lt;-str_remove(ref_trait, pattern=&quot;.txt&quot;)

    munged_trait&lt;-str_remove(all_files$munged_files[i], pattern=&quot;.sumstats.gz&quot;)
    print(ref_trait)
    print(munged_trait)
    
    if(munged_trait != ref_trait){print(&quot;Munged trait and reference trait don&#39;t match&quot;);break}else{print(&quot;Munged and reference input files match&quot;)}
    
    # read in file
    setwd(&quot;~/output/genetic_networks_project/gwas_munged/&quot;)
    munged_file&lt;-fread(as.character(all_files$munged_files[i]),header=T,data.table=F)
    
    # read in reference file
    setwd(&quot;~/output/regenie/step2/&quot;)
    ref_file&lt;-fread(as.character(all_files$ref_files[i]),header=T,data.table=F)
    
    # format ref file
    names(ref_file)[which(names(ref_file)==&quot;ID&quot;)]&lt;-&quot;SNP&quot;
    names(ref_file)[which(names(ref_file)==&quot;GENPOS&quot;)]&lt;-&quot;BP&quot;
    names(ref_file)[which(names(ref_file)==&quot;CHROM&quot;)]&lt;-&quot;CHR&quot;
    names(ref_file)[which(names(ref_file)==&quot;EAFREQ&quot;)]&lt;-&quot;EAF_prelim&quot;

    # only keep relevant columns from the reference file
    ref_file&lt;-ref_file[,c(&quot;BP&quot;,&quot;CHR&quot;,&quot;SNP&quot;,&quot;P&quot;,&quot;EA&quot;,&quot;EAF_prelim&quot;)]


    # merge munged and ref file
    merged_file&lt;-merge(munged_file, ref_file, by=c(&quot;SNP&quot;), all.x=T)


    # create an effect allele frequency column that matches the allele used in the munged file (because we take the frequency from the reference file that might have flipped effect and other alleles)
    merged_file$EAF&lt;-ifelse(merged_file$A1 == merged_file$EA, merged_file$EAF_prelim, 1-merged_file$EAF_prelim)

    # only keep required columns
    names(merged_file)[which(names(merged_file) == &quot;SNP&quot;)]&lt;-&quot;SNPID&quot;
    merged_file&lt;-merged_file[,c(&quot;SNPID&quot;,&quot;CHR&quot;,&quot;BP&quot;,&quot;A1&quot;,&quot;A2&quot;,&quot;EAF&quot;,&quot;N&quot;,&quot;Z&quot;,&quot;P&quot;)]
    
    # name according to GWAMA expectations
    # In genomicSEM A1 is the effect allele
    names(merged_file)[which(names(merged_file) == &quot;A1&quot;)]&lt;-&quot;EA&quot;
    names(merged_file)[which(names(merged_file) == &quot;A2&quot;)]&lt;-&quot;OA&quot;
    
    print(head(merged_file))
    
    ## print file
    setwd(&quot;~/output/genetic_networks_project/GWAMA_format/&quot;)
    fwrite(merged_file,file=paste0(&quot;gwama_format_&quot;,ref_trait), quote=FALSE,col.names=TRUE,row.names=F,sep=&quot; &quot;,na=&quot;NA&quot;)
}</code></pre>
</div>
<div id="create-pc-sumstats-weighted-with-loadings" class="section level2">
<h2>2. Create PC sumstats weighted with loadings</h2>
<pre class="r"><code>### calculate PC summary statistics for whole brain
## challenge make sure the right loadings weight the right file

# load dependencies
source(&quot;https://github.com/baselmans/multivariate_GWAMA/blob/master/Test_Data/N_weighted_GWAMA.function.1_2_6.R?raw=TRUE&quot;)
library(data.table)
library(stringr)

setwd(&quot;~/scripts/genetic_networks_project/gwama/&quot;)
source(&quot;my_GWAMA_26032020.R&quot;)

#### before reading in any files, create a data frame that contains matched file_names, Regions and loadings

# set wd to where GWAMA formatted files are saved
setwd(&quot;~/output/genetic_networks_project/GWAMA_format/&quot;)

# create vector with file_names
file_names&lt;-list.files(pattern=&quot;gwama_format_&quot;)

# create vector with Regions
Regions&lt;-str_remove(file_names,pattern=&quot;.txt&quot;)
Regions&lt;-str_remove(Regions,pattern=&quot;gwama_format_&quot;)


# combine file_names and Regions in one data.frame
df&lt;-data.frame(cbind(file_names,Regions))
df$file_names&lt;-as.character(df$file_names)
df$Regions&lt;-as.character(df$Regions)
str(df)

# load in loadings (they have been calculated with script: 5Eigendecomp/ eigen_decomposition_18052020.R, and transferred onto server)
setwd(&quot;~/data/standardised_loadings/&quot;)

networks&lt;-list.files(pattern=&quot;.txt&quot;)

# exclude whole brain, we are doing it separately because it needs so much memory
networks&lt;-networks[-which(networks == &quot;stand_loadings_whole_brain.txt&quot;)]

for(i in networks){
    print(i)
  loadings&lt;-fread(i,header=T,data.table=F)
  print(dim(loadings))
  name&lt;-str_replace(i,pattern=&quot;.txt&quot;,replacement = &quot;&quot;)
  assign(name,loadings)
}

loadings_all_traits&lt;-ls(pattern=&quot;stand_loadings_&quot;)
loadings_all_traits&lt;-loadings_all_traits[1:9]
print(loadings_all_traits)



###################### this will also be in the loop
# merge data frames for file_names,Regions,loadings_all_traits

for (i in loadings_all_traits){

    # retain only regions for this network that are stored in 1st column of each loadings_all_traits
    print(i)
    loadings&lt;-get(i)
    info&lt;-merge(loadings,df,by=&quot;Regions&quot;,all.x=T)
    print(&quot;This is the info data.frame&quot;)
    print(info) # for central exec we are expecting 8 nodes etc.
    
    # set wd to where GWAMA formatted files are stored
    setwd(&quot;~/output/genetic_networks_project/GWAMA_format/&quot;)

    # create list to hold GWASs
    num&lt;-nrow(info)
    dat&lt;-vector(&quot;list&quot;,num)

    # read in files for brain regions in this network
    print(&quot;Reading in GWAS files&quot;)
    for(j in 1:nrow(info)){
    print(j)
    file&lt;-info$file_names[j]
    dat_name&lt;-info$Regions[j]
    print(file)
    dat[[dat_name]]&lt;-fread(file,header=T,data.table=F)}
    dat[sapply(dat,is.null)]&lt;-NULL
    order&lt;-info$Regions
    dat&lt;-dat[order]
    print(str(dat))
    
    
    ####### read in intercepts
    print(&quot;Reading in intercepts&quot;)

    network_name&lt;-str_replace(i,pattern=&quot;stand_loadings_&quot;,replacement=&quot;&quot;)
    network_name&lt;-paste0(network_name,&quot;.RData&quot;)
    print(network_name)
    
    setwd(&quot;~/output/genetic_networks_project/ldsc/&quot;)
    intercept_file&lt;-list.files(pattern=network_name)
    #print(intercept_file)
    intercept_file&lt;-load(intercept_file)
    #print(intercept_file)
    
    # name rows and columns after their corresponding brain regions
    dimnames(LDSCoutput$I)[[1]]&lt;-dimnames(LDSCoutput$S)[[2]]
    dimnames(LDSCoutput$I)[[2]]&lt;-dimnames(LDSCoutput$S)[[2]]
    #print(LDSCoutput$I)
    
    # order intercept matrix in the same order as files were read in 
    print(order)
    intercepts&lt;-LDSCoutput$I[order,order]

    if(names(dat) != dimnames(intercepts)[[2]]){print(&quot;Intercepts are not in same order as files read in&quot;);break}
    print(intercepts)

    # store stand_loadings and name of the resulting log file in variable
    print(&quot;These are the standardised loadings for this network&quot;)
    stand_loadings&lt;-info$stand_loadings
    print(stand_loadings)
    
    print(&quot;This is the network:&quot;)
    name&lt;-str_replace(network_name,pattern=&quot;.RData&quot;,replacement=&quot;&quot;)
    print(name)
    
    # multiply the loadings with -1 for all networks if the median value of the loadings is below 0
    median_loadings&lt;-median(stand_loadings)
    mean_direction&lt;-sign(median_loadings)
    if(mean_direction == -1){
    stand_loadings&lt;-stand_loadings*(-1)
    } else {
    stand_loadings&lt;-stand_loadings}
    print(stand_loadings)

    setwd(&quot;~/output/genetic_networks_project/PC_sumstats/&quot;)
    print(&quot;Starting to create GWAMA scores&quot;)
                my_GWAMA(x=dat,
                          cov_Z=intercepts,
                          h2=stand_loadings,
                          out=&quot;.&quot;,
                          name=name,
                          output_gz=F,
                          check_columns=F)
        rm(dat)
        rm(LDSCoutput)
}
print(&quot;Done creating all GWAMA scores, excluding whole brain&quot;)</code></pre>
<p>Note that same script was used for whole brain, but submitted to the cluster separately because it needed more resources (the GWAMA function reads in all 83 files simultaneously).</p>
</div>
<div id="format-and-munge-pc-sumstats" class="section level2">
<h2>3. Format and munge PC sumstats</h2>
<p>This step is only performed the get the sumstats in shape, and no SNPs should be excluded because the GWAMA analyses were only performed on SNPs that survived the munging process in the first place.</p>
<pre class="r"><code>### Munge the PC_sumstats files and calculate ldsc with the cognitive GWAS files (already munged)

# load dependencies
library(tidyr)
library(stringr)
library(data.table)
library(devtools)
library(GenomicSEM)

# set wd to where formatted files are saved
setwd(&quot;~/output/genetic_networks_project/PC_sumstats&quot;)

## change header names for PC_sumstats

# read PC_files into R
PC_files&lt;-list.files(pattern=&quot;N_weighted_GWAMA.results.txt&quot;)
print(PC_files)

for (i in 1:length(PC_files)) assign(PC_files[i],fread(PC_files[i],header=T,data.table=F))
print(ls(pattern=&quot;N_weighted_GWAMA.results.txt&quot;))

# read in file that contains info columns
info_file&lt;-fread(&quot;~/output/regenie/step2/GWAS_22chr_noTBVcontrol_Right_transverse_temporal.txt&quot;,header=T,data.table=F)
names(info_file)[which(names(info_file)==&quot;CHROM&quot;)]&lt;-&quot;CHR&quot;
names(info_file)[which(names(info_file)==&quot;GENPOS&quot;)]&lt;-&quot;BP&quot;
names(info_file)[which(names(info_file)==&quot;ID&quot;)]&lt;-&quot;SNPID&quot;


# rename columns
for (i in PC_files){
        file&lt;-get(i)
        print(head(file))
        names(file)&lt;-c(&quot;CPTID&quot;,&quot;SNPID&quot;,&quot;CHR&quot;,&quot;BP&quot;,&quot;EA&quot;,&quot;OA&quot;,&quot;EAF&quot;,&quot;MAF&quot;,&quot;N&quot;,&quot;N_obs&quot;,&quot;Direction&quot;,&quot;BETA&quot;,&quot;SE&quot;,&quot;Z&quot;,&quot;P&quot;)
        print(head(file))
        file&lt;-file[,c(&quot;SNPID&quot;,&quot;CHR&quot;,&quot;BP&quot;,&quot;EA&quot;,&quot;OA&quot;,&quot;MAF&quot;,&quot;N&quot;,&quot;BETA&quot;,&quot;SE&quot;,&quot;P&quot;)]
        
        file&lt;-merge(info_file[,c(&quot;CHR&quot;,&quot;BP&quot;,&quot;SNPID&quot;,&quot;INFO&quot;)],file, by=c(&quot;CHR&quot;,&quot;BP&quot;,&quot;SNPID&quot;), all.y=T)
        print(head(file))
        
        
        setwd(&quot;~/output/genetic_networks_project/PC_header/&quot;)
        file_name&lt;-paste0(i,&quot;_ready_to_munge&quot;)
        file_name&lt;-str_remove(file_name,pattern=&quot;.N_weighted_GWAMA.results.txt&quot;)
        print(file_name)
        fwrite(file, file=file_name,quote=FALSE,col.names=TRUE,row.names=F,sep=&quot; &quot;,na=&quot;NA&quot;)
}
rm(list=ls())

######################## 
## munge PC sumstats that have been formatted in previous step

# load dependencies
library(tidyr)
library(stringr)
library(data.table)
library(devtools)
library(GenomicSEM)

hm3&lt;-&quot;~/data/w_hm3.noMHC.snplist&quot;

# list PC_sumstats files you want to munge
setwd(&quot;~/output/genetic_networks_project/PC_header/&quot;)
PC_files&lt;-list.files(pattern=&quot;_ready_to_munge&quot;)
print(length(PC_files))

# build a data frame that contains information on file_name and corresponding network

network_names&lt;-str_remove(PC_files,pattern=&quot;_ready_to_munge&quot;)


info&lt;-data.frame(cbind(PC_files,network_names))

info$PC_files&lt;-as.character(info$PC_files)
info$network_names&lt;-as.character(info$network_names)
print(str(info))


for (i in 1:nrow(info)){
        setwd(&quot;~/output/genetic_networks_project/PC_header/&quot;)
        files&lt;-info[i,&quot;PC_files&quot;]
        print(files)
        trait.names&lt;-info[i,&quot;network_names&quot;]
        print(trait.names)
        munge(files=files,
                hm3=hm3,
                trait.names=trait.names,
                info.filter = 0.9)
}</code></pre>
</div>
</div>
<div id="genetic-correlations-between-network-morphology-and-cognitive-traits" class="section level1">
<h1>Genetic correlations between network morphology and cognitive traits</h1>
<pre class="r"><code>#### run ldsc for cognitive traits and all networks included 

# load dependencies
library(tidyr)
library(stringr)
library(data.table)
library(devtools)
library(GenomicSEM)

# list munged PC sumstats
setwd(&quot;~/output/genetic_networks_project/PC_munged/&quot;)

# list files you would like to munge
PC_files&lt;-list.files(pattern=&quot;.sumstats.gz&quot;)
PC_files&lt;-paste0(&quot;~/output/genetic_networks_project/PC_munged/&quot;,PC_files)
print(PC_files)

# list cognitive traits
setwd(&quot;~/data/cognititve_GWAS&quot;)

cognitive_files&lt;-list.files(pattern=&quot;.sumstats.gz&quot;)

# specify traits used for ldsc
traits&lt;-append(PC_files,cognitive_files)
trait.names&lt;-str_replace(traits,pattern=&quot;.sumstats.gz&quot;,replacement=&quot;&quot;)
trait.names&lt;-str_remove(trait.names, pattern=&quot;~/output/genetic_networks_project/PC_munged/&quot;)

# ld scores and weights previously downloaded (European population)
ld&lt;-&quot;~/data/eur_w_ld_chr/&quot;
wld&lt;-&quot;~/data/eur_w_ld_chr/&quot;

# no sample or population prevalence required: NA for all traits
desired_length&lt;-length(traits)
sample.prev&lt;-rep(NA, desired_length)
population.prev&lt;-rep(NA, desired_length)

LDSCoutput&lt;-ldsc(traits=traits,
                    ld=ld,wld=wld,
                    trait.names = trait.names,
                    ldsc.log=&quot;~/output/genetic_networks_project/network_ldsc/ldsc_all_networks_cognitivetraits.log&quot;,
                    sample.prev=sample.prev,
                    population.prev=population.prev,
                    stand=T)
                    
save(LDSCoutput, file=&quot;~/output/genetic_networks_project/network_ldsc/all_networks_cognitive.RData&quot;)</code></pre>
<p><img src="Hypothesis3_files/figure-html/heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Note the small correlations of reaction time with the networks as well as the other cognitive traits.</p>
<p><br></p>
<div id="genetic-correlations-among-networks" class="section level3">
<h3>Genetic correlations among networks</h3>
<p>The genetic networks are very closely correlated, which would explain why there is no evidence for significantly different magnitudes of association between networks and general cognitive ability.</p>
<pre class="r"><code>temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc_networks/&quot;)
setwd(temporarywd)

load(&quot;all_networks_cognitive.RData&quot;)

ldscoutput&lt;-LDSCoutput
trait_names&lt;-dimnames(ldscoutput$S)[[2]]
dimnames(ldscoutput$S_Stand)[[1]]&lt;-trait_names
dimnames(ldscoutput$S_Stand)[[2]]&lt;-trait_names
ldscoutput$S_Stand&lt;-round(ldscoutput$S_Stand,digits = 2)
diag(ldscoutput$S_Stand)&lt;-NA

only_networks&lt;-ldscoutput$S_Stand[trait_names[1:10],trait_names[1:10]]
only_networks&lt;-get_lower_tri(only_networks)
only_networks&lt;-reshape2::melt(only_networks)

# only keep non-missing
# we only calculated 10*(10-1)/2=45, all others are redundant
only_networks&lt;-only_networks[which(!is.na(only_networks$value)),]

library(DT)
datatable(only_networks, caption = &quot;Genetic correlations among genetic networks&quot;,colnames = c(&quot;Network 1&quot;,&quot;Network 2&quot;,&quot;rg&quot;), rownames=FALSE,options = list(pageLength=5, scrollX=T))</code></pre>
<div id="htmlwidget-1922692a7d30286f7309" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1922692a7d30286f7309">{"x":{"filter":"none","caption":"<caption>Genetic correlations among genetic networks<\/caption>","data":[["cingulo","default_mode","hippocampal","multiple","p_fit","salience","sensori","temporo","whole_brain","default_mode","hippocampal","multiple","p_fit","salience","sensori","temporo","whole_brain","hippocampal","multiple","p_fit","salience","sensori","temporo","whole_brain","multiple","p_fit","salience","sensori","temporo","whole_brain","p_fit","salience","sensori","temporo","whole_brain","salience","sensori","temporo","whole_brain","sensori","temporo","whole_brain","temporo","whole_brain","whole_brain"],["central_exec","central_exec","central_exec","central_exec","central_exec","central_exec","central_exec","central_exec","central_exec","cingulo","cingulo","cingulo","cingulo","cingulo","cingulo","cingulo","cingulo","default_mode","default_mode","default_mode","default_mode","default_mode","default_mode","default_mode","hippocampal","hippocampal","hippocampal","hippocampal","hippocampal","hippocampal","multiple","multiple","multiple","multiple","multiple","p_fit","p_fit","p_fit","p_fit","salience","salience","salience","sensori","sensori","temporo"],[0.88,0.92,0.66,0.94,0.95,0.69,0.82,0.85,0.91,0.86,0.68,0.87,0.89,0.84,0.82,0.86,0.91,0.75,0.89,0.95,0.71,0.82,0.93,0.95,0.63,0.71,0.79,0.69,0.77,0.81,0.92,0.69,0.88,0.83,0.9,0.73,0.85,0.94,0.97,0.76,0.77,0.83,0.8,0.89,0.95]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Network 1<\/th>\n      <th>Network 2<\/th>\n      <th>rg<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p>The genetic correlations between the networks range from 0.97 to 0.63 (mean rg 0.83; SD = 0.09). This indicates that the different networks index very similar polygenic signal. The largest genetic correlation of 0.97 was between whole_brain and p_fit. The smallest genetic correlation of 0.63 was between multiple and hippocampal.</p>
<p><br> <br></p>
</div>
</div>
<div id="genomic-sem-genetic-intelligence-factor-g" class="section level1">
<h1>Genomic SEM: Genetic intelligence factor <em>g</em></h1>
<p>Here we assess the genetic correlations between network morphology and a factor of general cognitive ability <em>g</em> which is modeled from seven cognitive trait GWAS.</p>
<pre class="r"><code>#load dependencies

library(stringr)
library(data.table)
library(devtools)</code></pre>
<pre><code>## Error: package or namespace load failed for &#39;devtools&#39; in loadNamespace(j &lt;- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]):
##  namespace &#39;pkgload&#39; 1.1.0 is being loaded, but &gt;= 1.2.1 is required</code></pre>
<pre class="r"><code>library(GenomicSEM)

# load correlation matrix
temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc_networks/&quot;)
setwd(temporarywd)

load(&quot;all_networks_cognitive.RData&quot;)
ldscoutput&lt;-LDSCoutput

#name the traits accordingly
trait_names&lt;-dimnames(ldscoutput$S)[[2]]

dimnames(ldscoutput$S)[[1]]&lt;-trait_names
dimnames(ldscoutput$S)[[2]]&lt;-trait_names

# list all networks names
network_names&lt;-c(&quot;central_exec&quot;,&quot;cingulo&quot;,&quot;default_mode&quot;,&quot;hippocampal&quot;,&quot;multiple&quot;,&quot;p_fit&quot;,&quot;salience&quot;,&quot;sensori&quot;,&quot;temporo&quot;,&quot;whole_brain&quot;)

#### genomic SEM

for (i in network_names){
            
            # specify covariance structure of interest
            covstruc &lt;- ldscoutput

            #specify estimation method
            estimation&lt;- &quot;DWLS&quot;

            # specify your SEM
            network_name&lt;-i
            #PC_model&lt;- &#39;G =~ NA*Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR
            #           G ~~ 1*G 
            #           G ~ central_exec&#39;
            
            PC_model&lt;-paste0(&quot;G =~ NA*Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR\n\t\t\t\t\t\tG ~~ 1*G \n\t\t\t\t\t\tG ~ &quot;,network_name)

            # run your model    
            SEM_model&lt;-usermodel(covstruc=covstruc, model=PC_model, estimation=estimation, imp_cov=T)
            model_name&lt;-paste0(network_name,&quot;_cognitive_SEM_results&quot;)
            assign(model_name,SEM_model)
}
results&lt;-ls(pattern=&quot;_cognitive_SEM_results&quot;)</code></pre>
<div id="model-fit-for-associations-with-genetic-g" class="section level3">
<h3>Model fit for associations with genetic <em>g</em></h3>
<pre class="r"><code>library(knitr)

fit_table&lt;-NULL

for (i in results){
    SEM_results&lt;-get(i)
    fit_table&lt;-data.frame(rbind(fit_table,SEM_results$modelfit))
}

model&lt;-str_replace(results,pattern=&quot;_SEM_results&quot;,replacement=&quot;&quot;)
fit_table$model&lt;-model
#fit_table$p_chisq&lt;-signif(fit_table$p_chisq)
fit_table&lt;-fit_table[,c(&quot;model&quot;,&quot;chisq&quot;,&quot;df&quot;,&quot;p_chisq&quot;,&quot;AIC&quot;,&quot;CFI&quot;,&quot;SRMR&quot;)]
kable(fit_table,caption=&quot;Model fit across models&quot;,format=&quot;markdown&quot;,digits = c(0,2,0,24,2,2,2),row.names = F)</code></pre>
<table>
<caption>Model fit across models</caption>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="right">chisq</th>
<th align="right">df</th>
<th align="right">p_chisq</th>
<th align="right">AIC</th>
<th align="right">CFI</th>
<th align="right">SRMR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">central_exec_cognitive</td>
<td align="right">142.04</td>
<td align="right">20</td>
<td align="right">2.0726e-20</td>
<td align="right">174.04</td>
<td align="right">0.97</td>
<td align="right">0.08</td>
</tr>
<tr class="even">
<td align="left">cingulo_cognitive</td>
<td align="right">156.50</td>
<td align="right">20</td>
<td align="right">3.6000e-23</td>
<td align="right">188.50</td>
<td align="right">0.96</td>
<td align="right">0.08</td>
</tr>
<tr class="odd">
<td align="left">default_mode_cognitive</td>
<td align="right">150.83</td>
<td align="right">20</td>
<td align="right">4.3600e-22</td>
<td align="right">182.83</td>
<td align="right">0.96</td>
<td align="right">0.08</td>
</tr>
<tr class="even">
<td align="left">hippocampal_cognitive</td>
<td align="right">143.69</td>
<td align="right">20</td>
<td align="right">1.0088e-20</td>
<td align="right">175.69</td>
<td align="right">0.96</td>
<td align="right">0.08</td>
</tr>
<tr class="odd">
<td align="left">multiple_cognitive</td>
<td align="right">143.06</td>
<td align="right">20</td>
<td align="right">1.3289e-20</td>
<td align="right">175.06</td>
<td align="right">0.97</td>
<td align="right">0.08</td>
</tr>
<tr class="even">
<td align="left">p_fit_cognitive</td>
<td align="right">161.08</td>
<td align="right">20</td>
<td align="right">5.0000e-24</td>
<td align="right">193.08</td>
<td align="right">0.96</td>
<td align="right">0.08</td>
</tr>
<tr class="odd">
<td align="left">salience_cognitive</td>
<td align="right">144.92</td>
<td align="right">20</td>
<td align="right">5.8740e-21</td>
<td align="right">176.92</td>
<td align="right">0.96</td>
<td align="right">0.08</td>
</tr>
<tr class="even">
<td align="left">sensori_cognitive</td>
<td align="right">146.20</td>
<td align="right">20</td>
<td align="right">3.3420e-21</td>
<td align="right">178.20</td>
<td align="right">0.96</td>
<td align="right">0.08</td>
</tr>
<tr class="odd">
<td align="left">temporo_cognitive</td>
<td align="right">165.98</td>
<td align="right">20</td>
<td align="right">1.0000e-24</td>
<td align="right">197.98</td>
<td align="right">0.96</td>
<td align="right">0.08</td>
</tr>
<tr class="even">
<td align="left">whole_brain_cognitive</td>
<td align="right">157.75</td>
<td align="right">20</td>
<td align="right">2.0000e-23</td>
<td align="right">189.75</td>
<td align="right">0.96</td>
<td align="right">0.08</td>
</tr>
</tbody>
</table>
</div>
<div id="genetic-correlation-between-network-morphology-and-a-genetic-g-factor" class="section level2">
<h2>Genetic correlation between network morphology and a genetic <em>g</em> factor</h2>
<pre class="r"><code>network_assoc&lt;-NULL

for(i in results){
    SEM_results&lt;-get(i)
    network_assoc&lt;-rbind(network_assoc,SEM_results$results[2,c(&quot;STD_Genotype&quot;,&quot;STD_Genotype_SE&quot;,&quot;p_value&quot;)])
}

genetic_cor&lt;-data.frame(cbind(model,network_assoc))

##create CI column
genetic_cor$STD_Genotype_SE&lt;-as.numeric(genetic_cor$STD_Genotype_SE)
genetic_cor$ci_u&lt;-genetic_cor$STD_Genotype-(1.96*genetic_cor$STD_Genotype_SE)
genetic_cor$ci_l&lt;-genetic_cor$STD_Genotype+(1.96*genetic_cor$STD_Genotype_SE)

# adjust for multiple testing
genetic_cor$p_adjust&lt;-p.adjust(genetic_cor$p_value,method=&quot;fdr&quot;,n=nrow(genetic_cor))

# round your numbers
#genetic_cor&lt;-data.frame(lapply(genetic_cor, function(y) if(is.numeric(y)) signif(y) else y))
genetic_cor$STD_Genotype&lt;-round(genetic_cor$STD_Genotype,digits = 2)
genetic_cor$STD_Genotype_SE&lt;-round(genetic_cor$STD_Genotype_SE, digits = 3)
genetic_cor$p_value&lt;-signif(genetic_cor$p_value)
genetic_cor$p_adjust&lt;-signif(genetic_cor$p_adjust)
genetic_cor$ci_l&lt;-round(genetic_cor$ci_l,digits = 2)
genetic_cor$ci_u&lt;-round(genetic_cor$ci_u,digits = 2)

genetic_cor$CI&lt;-with(genetic_cor, paste(ci_u,ci_l,sep = &quot; - &quot;))

genetic_cor&lt;-genetic_cor[,c(&quot;model&quot;,&quot;STD_Genotype&quot;,&quot;STD_Genotype_SE&quot;,&quot;CI&quot;,&quot;p_value&quot;,&quot;p_adjust&quot;)]
kable(genetic_cor,caption=&quot;Genetic correlations between PC sumstats and genetic g&quot;,format=&quot;markdown&quot;, row.names = F)</code></pre>
<table>
<caption>Genetic correlations between PC sumstats and genetic g</caption>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="right">STD_Genotype</th>
<th align="right">STD_Genotype_SE</th>
<th align="left">CI</th>
<th align="right">p_value</th>
<th align="right">p_adjust</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">central_exec_cognitive</td>
<td align="right">0.20</td>
<td align="right">0.037</td>
<td align="left">0.12 - 0.27</td>
<td align="right">1.00e-07</td>
<td align="right">3.00e-07</td>
</tr>
<tr class="even">
<td align="left">cingulo_cognitive</td>
<td align="right">0.20</td>
<td align="right">0.038</td>
<td align="left">0.13 - 0.27</td>
<td align="right">1.00e-07</td>
<td align="right">3.00e-07</td>
</tr>
<tr class="odd">
<td align="left">default_mode_cognitive</td>
<td align="right">0.19</td>
<td align="right">0.037</td>
<td align="left">0.12 - 0.26</td>
<td align="right">2.00e-07</td>
<td align="right">3.00e-07</td>
</tr>
<tr class="even">
<td align="left">hippocampal_cognitive</td>
<td align="right">0.17</td>
<td align="right">0.039</td>
<td align="left">0.09 - 0.24</td>
<td align="right">2.66e-05</td>
<td align="right">2.66e-05</td>
</tr>
<tr class="odd">
<td align="left">multiple_cognitive</td>
<td align="right">0.19</td>
<td align="right">0.039</td>
<td align="left">0.12 - 0.27</td>
<td align="right">7.00e-07</td>
<td align="right">9.00e-07</td>
</tr>
<tr class="even">
<td align="left">p_fit_cognitive</td>
<td align="right">0.20</td>
<td align="right">0.038</td>
<td align="left">0.12 - 0.27</td>
<td align="right">2.00e-07</td>
<td align="right">3.00e-07</td>
</tr>
<tr class="odd">
<td align="left">salience_cognitive</td>
<td align="right">0.19</td>
<td align="right">0.036</td>
<td align="left">0.12 - 0.26</td>
<td align="right">3.00e-07</td>
<td align="right">4.00e-07</td>
</tr>
<tr class="even">
<td align="left">sensori_cognitive</td>
<td align="right">0.19</td>
<td align="right">0.039</td>
<td align="left">0.11 - 0.27</td>
<td align="right">1.20e-06</td>
<td align="right">1.30e-06</td>
</tr>
<tr class="odd">
<td align="left">temporo_cognitive</td>
<td align="right">0.20</td>
<td align="right">0.038</td>
<td align="left">0.12 - 0.27</td>
<td align="right">2.00e-07</td>
<td align="right">4.00e-07</td>
</tr>
<tr class="even">
<td align="left">whole_brain_cognitive</td>
<td align="right">0.21</td>
<td align="right">0.039</td>
<td align="left">0.13 - 0.29</td>
<td align="right">1.00e-07</td>
<td align="right">3.00e-07</td>
</tr>
</tbody>
</table>
<p>Note that the adjusted <em>p</em>-values were adjusted using a 5% false discovery rate.</p>
<p><img src="Path_diagram_central_exec.jpg" /></p>
<p><br> <br> <br></p>
</div>
</div>
<div id="compare-networks-based-on-their-correlation-magnitudes-with-general-cognitive-abilities" class="section level1">
<h1>Compare networks based on their correlation magnitudes with general cognitive abilities</h1>
<p>Next, we build SEMs to understand whether magnitudes between genetic <em>g</em> and PC sumstats for the Central Executive network are significantly different from the associations between genetic g and the other networks. We do this for raw correlations displayed above, but also for correlations adjusted for the number of nodes contributing to individual networks.</p>
<p><img src="Path_diagram_compare_corr.jpg" /></p>
<p><br></p>
<hr />
<div id="raw-correlations" class="section level2">
<h2>Raw correlations</h2>
<pre class="r"><code>library(stringr)
library(data.table)
#library(devtools)
library(GenomicSEM)

temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc_networks/&quot;)
setwd(temporarywd)

load(&quot;all_networks_cognitive.RData&quot;)
ldscoutput&lt;-LDSCoutput

#### genomic SEM
# compare central_exec to all other networks
other_networks&lt;-c(&quot;cingulo&quot;,&quot;default_mode&quot;,&quot;hippocampal&quot;,&quot;multiple&quot;,&quot;p_fit&quot;,&quot;salience&quot;,&quot;sensori&quot;,&quot;temporo&quot;,&quot;whole_brain&quot;)

# run constrained model
for(i in other_networks){

            # specify covariance structure of interest
            dimnames(LDSCoutput$S)[[2]]&lt;-dimnames(LDSCoutput$S)[[2]]
            covstruc &lt;- LDSCoutput

            #specify estimation method
            estimation&lt;- &quot;DWLS&quot;
            
            # specify your constrained model
            constrained&lt;-paste0(&quot;G =~ NA*Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR
                                G ~~ 1*G 
                                G ~ a*central_exec
                                G ~ a*&quot;,i,&quot;
                                central_exec ~~ &quot;,i) 

            # run your constrained model    
            constrained_model&lt;-usermodel(covstruc=covstruc, model=constrained, estimation=estimation, imp_cov=T)

            # save the model under appropriate name
            model_name&lt;-paste0(&quot;constrained_model_central_exec_&quot;,i)
            assign(model_name,constrained_model)


            # specify your freely-varying model
            free&lt;-paste0(&quot;G =~ NA*Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR
                                G ~~ 1*G 
                                G ~ central_exec
                                G ~ &quot;,i,&quot;
                                central_exec ~~ &quot;,i)            
            
            free_model&lt;-usermodel(covstruc=covstruc, model=free, estimation=estimation, imp_cov=T)
            
            model_name&lt;-paste0(&quot;free_model_central_exec_&quot;,i)
            assign(model_name,free_model)
            }

results&lt;-ls(pattern=&quot;_model_central_exec_&quot;)</code></pre>
<div id="model-fit-constrained-vs.-freely-varying-models" class="section level3">
<h3>Model fit constrained vs. freely varying models</h3>
<pre class="r"><code>constrained_results&lt;-ls(pattern=&quot;constrained_model_central_exec_&quot;)

table&lt;-data.frame()
for (i in constrained_results){
    results&lt;-get(i)
    names(results$modelfit)&lt;-paste0(&quot;constrained_&quot;,names(results$modelfit))
    table&lt;-rbind(table,results$modelfit)
    assign(&quot;constrained_table&quot;,table)
    }
constrained_table$model&lt;-str_replace(constrained_results,pattern=&quot;constrained_model_&quot;,replacement=&quot;&quot;)


free_results&lt;-ls(pattern=&quot;free_model_central_exec_&quot;)

table&lt;-data.frame()
for (i in free_results){
    results&lt;-get(i)
    names(results$modelfit)&lt;-paste0(&quot;free_&quot;,names(results$modelfit))
    table&lt;-rbind(table,results$modelfit)
    assign(&quot;free_table&quot;,table)
    }

free_table$model&lt;-str_replace(free_results,pattern=&quot;free_model_&quot;,replacement=&quot;&quot;)

# differences in chi-squared are chi-squared distributed with differences in degrees of freedom and can be checked manually for significance using a chi-squared table
#https://www.psychologie.uzh.ch/dam/jcr:ffffffff-b371-2797-0000-00000fda8f29/chisquare_diff_en.pdf
model_fit_all&lt;-merge(constrained_table,free_table,by=&quot;model&quot;)
model_fit_all$delta_chisq&lt;-with(model_fit_all, constrained_chisq-free_chisq)
model_fit_all$delta_df&lt;-with(model_fit_all, constrained_df-free_df)
model_fit_all$chi_comp_p&lt;-with(model_fit_all, 1-pchisq(delta_chisq,delta_df))
model_fit_all$comp_p_adjust&lt;-p.adjust(model_fit_all$chi_comp_p,method=&quot;fdr&quot;,n=nrow(model_fit_all))

display_chi&lt;-model_fit_all[,c(&quot;model&quot;,&quot;delta_chisq&quot;,&quot;delta_df&quot;,&quot;chi_comp_p&quot;,&quot;comp_p_adjust&quot;)]
display_fit_other&lt;-model_fit_all[,c(&quot;model&quot;,&quot;constrained_AIC&quot;,&quot;constrained_CFI&quot;,&quot;constrained_SRMR&quot;,&quot;free_AIC&quot;,&quot;free_CFI&quot;,&quot;free_SRMR&quot;)]
kable(display_chi,caption=&quot;Significance test of difference in chi-squared values obtained as model fit in constrained vs. freely varying correlations&quot;,format=&quot;markdown&quot;,digits = c(0,3,0,3,3))</code></pre>
<table>
<caption>Significance test of difference in chi-squared values obtained as model fit in constrained vs. freely varying correlations</caption>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="right">delta_chisq</th>
<th align="right">delta_df</th>
<th align="right">chi_comp_p</th>
<th align="right">comp_p_adjust</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">central_exec_cingulo</td>
<td align="right">0.833</td>
<td align="right">1</td>
<td align="right">0.361</td>
<td align="right">0.675</td>
</tr>
<tr class="even">
<td align="left">central_exec_default_mode</td>
<td align="right">-0.173</td>
<td align="right">1</td>
<td align="right">1.000</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">central_exec_hippocampal</td>
<td align="right">1.102</td>
<td align="right">1</td>
<td align="right">0.294</td>
<td align="right">0.675</td>
</tr>
<tr class="even">
<td align="left">central_exec_multiple</td>
<td align="right">-0.584</td>
<td align="right">1</td>
<td align="right">1.000</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">central_exec_p_fit</td>
<td align="right">0.786</td>
<td align="right">1</td>
<td align="right">0.375</td>
<td align="right">0.675</td>
</tr>
<tr class="even">
<td align="left">central_exec_salience</td>
<td align="right">0.046</td>
<td align="right">1</td>
<td align="right">0.831</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">central_exec_sensori</td>
<td align="right">0.036</td>
<td align="right">1</td>
<td align="right">0.850</td>
<td align="right">1.000</td>
</tr>
<tr class="even">
<td align="left">central_exec_temporo</td>
<td align="right">0.955</td>
<td align="right">1</td>
<td align="right">0.328</td>
<td align="right">0.675</td>
</tr>
<tr class="odd">
<td align="left">central_exec_whole_brain</td>
<td align="right">3.226</td>
<td align="right">1</td>
<td align="right">0.072</td>
<td align="right">0.652</td>
</tr>
</tbody>
</table>
<pre class="r"><code>kable(display_fit_other,caption = &quot;Other fit indices of constrained vs. freely varying correlations&quot;,digits = 2)</code></pre>
<table style="width:100%;">
<caption>Other fit indices of constrained vs. freely varying correlations</caption>
<colgroup>
<col width="25%" />
<col width="15%" />
<col width="15%" />
<col width="16%" />
<col width="8%" />
<col width="8%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="right">constrained_AIC</th>
<th align="right">constrained_CFI</th>
<th align="right">constrained_SRMR</th>
<th align="right">free_AIC</th>
<th align="right">free_CFI</th>
<th align="right">free_SRMR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">central_exec_cingulo</td>
<td align="right">209.98</td>
<td align="right">0.99</td>
<td align="right">0.07</td>
<td align="right">211.15</td>
<td align="right">0.99</td>
<td align="right">0.07</td>
</tr>
<tr class="even">
<td align="left">central_exec_default_mode</td>
<td align="right">198.20</td>
<td align="right">0.99</td>
<td align="right">0.08</td>
<td align="right">200.37</td>
<td align="right">0.99</td>
<td align="right">0.08</td>
</tr>
<tr class="odd">
<td align="left">central_exec_hippocampal</td>
<td align="right">201.57</td>
<td align="right">0.97</td>
<td align="right">0.07</td>
<td align="right">202.47</td>
<td align="right">0.97</td>
<td align="right">0.07</td>
</tr>
<tr class="even">
<td align="left">central_exec_multiple</td>
<td align="right">199.32</td>
<td align="right">0.99</td>
<td align="right">0.07</td>
<td align="right">201.91</td>
<td align="right">0.99</td>
<td align="right">0.07</td>
</tr>
<tr class="odd">
<td align="left">central_exec_p_fit</td>
<td align="right">217.69</td>
<td align="right">1.00</td>
<td align="right">0.08</td>
<td align="right">218.90</td>
<td align="right">1.00</td>
<td align="right">0.08</td>
</tr>
<tr class="even">
<td align="left">central_exec_salience</td>
<td align="right">214.19</td>
<td align="right">0.98</td>
<td align="right">0.07</td>
<td align="right">216.15</td>
<td align="right">0.98</td>
<td align="right">0.07</td>
</tr>
<tr class="odd">
<td align="left">central_exec_sensori</td>
<td align="right">204.42</td>
<td align="right">0.98</td>
<td align="right">0.07</td>
<td align="right">206.39</td>
<td align="right">0.98</td>
<td align="right">0.07</td>
</tr>
<tr class="even">
<td align="left">central_exec_temporo</td>
<td align="right">213.98</td>
<td align="right">0.99</td>
<td align="right">0.08</td>
<td align="right">215.03</td>
<td align="right">0.99</td>
<td align="right">0.08</td>
</tr>
<tr class="odd">
<td align="left">central_exec_whole_brain</td>
<td align="right">218.24</td>
<td align="right">0.99</td>
<td align="right">0.07</td>
<td align="right">217.02</td>
<td align="right">0.99</td>
<td align="right">0.07</td>
</tr>
</tbody>
</table>
<p><br> <br></p>
<hr />
</div>
</div>
<div id="adjusted-correlations" class="section level2">
<h2>Adjusted correlations</h2>
<p>To account for different sizes of the networks (i.e. different number of included brain volumes), we compare whether associations with <em>g</em> differ between networks, when the associations have been divided by their number of included volumes.</p>
<pre class="r"><code>library(stringr)
library(data.table)
#library(devtools)
library(GenomicSEM)

temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc_networks/&quot;)
setwd(temporarywd)

load(&quot;all_networks_cognitive.RData&quot;)

#### genomic SEM
# compare central_exec to all other networks
other_networks&lt;-c(&quot;cingulo&quot;,&quot;default_mode&quot;,&quot;hippocampal&quot;,&quot;multiple&quot;,&quot;p_fit&quot;,&quot;salience&quot;,&quot;sensori&quot;,&quot;temporo&quot;,&quot;whole_brain&quot;)

# run constrained and freely varying model
for(i in other_networks){

            # specify covariance structure of interest
            var_names&lt;-str_replace(dimnames(LDSCoutput$S)[[2]],pattern=&quot;~/data/PC_munged/&quot;,replacement=&quot;&quot;)
            dimnames(LDSCoutput$S)[[2]]&lt;-var_names
            covstruc &lt;- LDSCoutput

            #specify estimation method
            # I would have thought we want to use ML but
            # consider Javiers explanation
            estimation&lt;- &quot;DWLS&quot;

            
            # specify number of nodes for each network to adjust the correlation
            if(i == &quot;cingulo&quot; | i == &quot;salience&quot;){
            nodes=10}
                else if(i == &quot;default_mode&quot;){
                nodes=16}
                    else if(i == &quot;hippocampal&quot; | i ==&quot;multiple&quot; | i == &quot;sensori&quot;){
                    nodes=12}
                        else if(i == &quot;p_fit&quot;){
                        nodes=36}
                            else if(i == &quot;temporo&quot;){
                            nodes=30}
                      else if(i == &quot;whole_brain&quot;){
                        nodes=83}
                                  else{break}
            print(i)
            print(nodes)
            
            
            
            # specify your constrained model
            constrained&lt;-paste0(&quot;G =~ NA*Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR\nG ~~ 1*G\nG ~ slope_central*central_exec\nG ~ slope_other*&quot;,i,&quot;\nslope_central/8 == slope_other/&quot;,nodes,&quot;\ncentral_exec ~~ &quot;,i)
            #constrained&lt;-paste0(&quot;G =~ NA*Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR
            #                   G ~~ 1*G
            #                   G ~ slope_central*central_exec
            #                   G ~ slope_other*&quot;,i,&quot;
            #                   slope_central/8 = slope_other/&quot;,nodes,&quot;
            #                   central_exec ~~ &quot;,i)
            print(constrained)
            
            # run your contsrained model    
            constrained_model&lt;-usermodel(covstruc=covstruc, model=constrained, estimation=estimation, imp_cov=T)

            model_name&lt;-paste0(&quot;constrained_model_central_exec_&quot;,i)
            assign(model_name,constrained_model)
            
            
            # specify your freely-varying model
            free&lt;-paste0(&quot;G =~ NA*Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR
                                G ~~ 1*G 
                                G ~ central_exec
                                G ~ &quot;,i,&quot;
                                central_exec ~~ &quot;,i)            
            
            free_model&lt;-usermodel(covstruc=covstruc, model=free, estimation=estimation, imp_cov=T)
            
            model_name&lt;-paste0(&quot;free_model_central_exec_&quot;,i)
            assign(model_name,free_model)
            }
            
results&lt;-ls(pattern=&quot;_model_central_exec_&quot;)</code></pre>
<div id="model-fit-constrained-vs.-freely-varying-models-1" class="section level3">
<h3>Model fit constrained vs. freely varying models</h3>
<pre class="r"><code>constrained_results&lt;-ls(pattern=&quot;constrained_model_central_exec_&quot;)

table&lt;-data.frame()
for (i in constrained_results){
    results&lt;-get(i)
    names(results$modelfit)&lt;-paste0(&quot;constrained_&quot;,names(results$modelfit))
    table&lt;-rbind(table,results$modelfit)
    assign(&quot;constrained_table&quot;,table)
    }
constrained_table$model&lt;-str_replace(constrained_results,pattern=&quot;constrained_model_&quot;,replacement=&quot;&quot;)


free_results&lt;-ls(pattern=&quot;free_model_central_exec_&quot;)

table&lt;-data.frame()
for (i in free_results){
    results&lt;-get(i)
    names(results$modelfit)&lt;-paste0(&quot;free_&quot;,names(results$modelfit))
    table&lt;-rbind(table,results$modelfit)
    assign(&quot;free_table&quot;,table)
    }

free_table$model&lt;-str_replace(free_results,pattern=&quot;free_model_&quot;,replacement=&quot;&quot;)

# combine constrained and free models
model_fit_all&lt;-merge(constrained_table,free_table,by=&quot;model&quot;)

# differences in chi-squared are chi-squared distributed with differences in degrees of freedom
#https://www.psychologie.uzh.ch/dam/jcr:ffffffff-b371-2797-0000-00000fda8f29/chisquare_diff_en.pdf
model_fit_all&lt;-merge(constrained_table,free_table,by=&quot;model&quot;)
model_fit_all$delta_chisq&lt;-with(model_fit_all, constrained_chisq-free_chisq)
model_fit_all$delta_df&lt;-with(model_fit_all, constrained_df-free_df)
model_fit_all$chi_comp_p&lt;-with(model_fit_all, 1-pchisq(delta_chisq,delta_df))
model_fit_all$comp_p_adjust&lt;-p.adjust(model_fit_all$chi_comp_p,method=&quot;fdr&quot;,n=nrow(model_fit_all))

display_chi&lt;-model_fit_all[,c(&quot;model&quot;,&quot;delta_chisq&quot;,&quot;delta_df&quot;,&quot;chi_comp_p&quot;,&quot;comp_p_adjust&quot;)]
display_fit_other&lt;-model_fit_all[,c(&quot;model&quot;,&quot;constrained_AIC&quot;,&quot;free_AIC&quot;,&quot;constrained_CFI&quot;,&quot;free_CFI&quot;,&quot;constrained_SRMR&quot;,&quot;free_SRMR&quot;)]
kable(display_chi,caption=&quot;Significance test of difference in chi-squared values obtained as model fit in constrained vs. freely varying correlations&quot;,format=&quot;markdown&quot;,digits = c(0,2,0,2,2))</code></pre>
<table>
<caption>Significance test of difference in chi-squared values obtained as model fit in constrained vs. freely varying correlations</caption>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="right">delta_chisq</th>
<th align="right">delta_df</th>
<th align="right">chi_comp_p</th>
<th align="right">comp_p_adjust</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">central_exec_cingulo</td>
<td align="right">0.72</td>
<td align="right">1</td>
<td align="right">0.40</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">central_exec_default_mode</td>
<td align="right">-1.30</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">central_exec_hippocampal</td>
<td align="right">1.60</td>
<td align="right">1</td>
<td align="right">0.21</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">central_exec_multiple</td>
<td align="right">-1.01</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">central_exec_p_fit</td>
<td align="right">-2.65</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">central_exec_salience</td>
<td align="right">-0.04</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">central_exec_sensori</td>
<td align="right">0.64</td>
<td align="right">1</td>
<td align="right">0.42</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">central_exec_temporo</td>
<td align="right">-1.61</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">central_exec_whole_brain</td>
<td align="right">-0.44</td>
<td align="right">1</td>
<td align="right">1.00</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<pre class="r"><code>kable(display_fit_other,caption = &quot;Other fit indices of constrained vs. freely varying correlations&quot;,digits=3)</code></pre>
<table style="width:100%;">
<caption>Other fit indices of constrained vs. freely varying correlations</caption>
<colgroup>
<col width="25%" />
<col width="15%" />
<col width="8%" />
<col width="15%" />
<col width="8%" />
<col width="16%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="right">constrained_AIC</th>
<th align="right">free_AIC</th>
<th align="right">constrained_CFI</th>
<th align="right">free_CFI</th>
<th align="right">constrained_SRMR</th>
<th align="right">free_SRMR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">central_exec_cingulo</td>
<td align="right">209.868</td>
<td align="right">211.148</td>
<td align="right">0.988</td>
<td align="right">0.988</td>
<td align="right">0.073</td>
<td align="right">0.074</td>
</tr>
<tr class="even">
<td align="left">central_exec_default_mode</td>
<td align="right">197.076</td>
<td align="right">200.375</td>
<td align="right">0.994</td>
<td align="right">0.994</td>
<td align="right">0.075</td>
<td align="right">0.075</td>
</tr>
<tr class="odd">
<td align="left">central_exec_hippocampal</td>
<td align="right">202.065</td>
<td align="right">202.466</td>
<td align="right">0.974</td>
<td align="right">0.974</td>
<td align="right">0.075</td>
<td align="right">0.074</td>
</tr>
<tr class="even">
<td align="left">central_exec_multiple</td>
<td align="right">198.893</td>
<td align="right">201.906</td>
<td align="right">0.993</td>
<td align="right">0.993</td>
<td align="right">0.073</td>
<td align="right">0.073</td>
</tr>
<tr class="odd">
<td align="left">central_exec_p_fit</td>
<td align="right">214.258</td>
<td align="right">218.904</td>
<td align="right">0.995</td>
<td align="right">0.995</td>
<td align="right">0.075</td>
<td align="right">0.075</td>
</tr>
<tr class="even">
<td align="left">central_exec_salience</td>
<td align="right">214.107</td>
<td align="right">216.149</td>
<td align="right">0.976</td>
<td align="right">0.976</td>
<td align="right">0.074</td>
<td align="right">0.073</td>
</tr>
<tr class="odd">
<td align="left">central_exec_sensori</td>
<td align="right">205.032</td>
<td align="right">206.388</td>
<td align="right">0.982</td>
<td align="right">0.982</td>
<td align="right">0.073</td>
<td align="right">0.072</td>
</tr>
<tr class="even">
<td align="left">central_exec_temporo</td>
<td align="right">211.416</td>
<td align="right">215.028</td>
<td align="right">0.988</td>
<td align="right">0.988</td>
<td align="right">0.077</td>
<td align="right">0.077</td>
</tr>
<tr class="odd">
<td align="left">central_exec_whole_brain</td>
<td align="right">214.577</td>
<td align="right">217.016</td>
<td align="right">0.994</td>
<td align="right">0.994</td>
<td align="right">0.075</td>
<td align="right">0.075</td>
</tr>
</tbody>
</table>
<p><br> <br></p>
</div>
</div>
<div id="calculate-genetic-correlations-between-central-executive-network-and-specific-cognitive-abilities" class="section level2">
<h2>Calculate genetic correlations between central executive network and specific cognitive abilities</h2>
<pre class="r"><code>library(stringr)
library(data.table)
#library(devtools)
library(GenomicSEM)

temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc_networks/&quot;)
setwd(temporarywd)

load(&quot;all_networks_cognitive.RData&quot;)


#### genomic SEM
# what&#39;s the association between all networks and three cognitive abilities

cognitive_abilities&lt;-c(&quot;Matrix&quot;,&quot;Symbol Digit&quot;,&quot;Memory&quot;)
networks&lt;-c(&quot;central_exec&quot;,&quot;cingulo&quot;,&quot;default_mode&quot;,&quot;hippocampal&quot;,&quot;multiple&quot;,&quot;p_fit&quot;,&quot;salience&quot;,&quot;sensori&quot;,&quot;temporo&quot;,&quot;whole_brain&quot;)


for(i in networks){
  for(j in cognitive_abilities){

    # covariance strutcure of interest
    covstruc &lt;- LDSCoutput
    
    # estimation method
    estimation&lt;- &quot;DWLS&quot;
    
    #specify model
    model&lt;-paste0(i,&quot; ~~ &quot;,j)
    
    # run model
    SEM_model&lt;-usermodel(covstruc=covstruc, model=model, estimation=estimation, imp_cov=T)
    
    # name and save model
    model_name&lt;-paste0(i,&quot;_&quot;,j,&quot;_indivresults&quot;)
    assign(model_name,SEM_model)
}}

results&lt;-ls(pattern=&quot;_indivresults&quot;)</code></pre>
<p><em>There are no fit statistics and zero degrees of freedom because the models are fully saturated.</em></p>
<pre class="r"><code>table&lt;-data.frame()
for (i in results){
    cor_results&lt;-get(i)
    table&lt;-rbind(table,cor_results$results[2,c(&quot;STD_Genotype&quot;,&quot;STD_Genotype_SE&quot;,&quot;p_value&quot;)])
}
model&lt;-str_replace(results,pattern=&quot;_results&quot;,replacement=&quot;&quot;)
table$model&lt;-model

##create CI column
table$STD_Genotype_SE&lt;-as.numeric(table$STD_Genotype_SE)

table$ci_u&lt;-table$STD_Genotype-(1.96*table$STD_Genotype_SE)
table$ci_l&lt;-table$STD_Genotype+(1.96*table$STD_Genotype_SE)

#adjust for multiple testing
table$p_adjust&lt;-p.adjust(table$p_value,method=&quot;fdr&quot;,n=nrow(table))

# round numbers
table&lt;-data.frame(lapply(table, function(y) if(is.numeric(y)) round(y, 4) else y))
# format CI column
table$CI&lt;-with(table, paste(ci_u,ci_l,sep = &quot; - &quot;))

table&lt;-table[,c(&quot;model&quot;,&quot;STD_Genotype&quot;,&quot;STD_Genotype_SE&quot;,&quot;CI&quot;,&quot;p_value&quot;,&quot;p_adjust&quot;)]

# make model name shorter, delete redundatn info
library(stringr)
table$model&lt;-str_remove(table$model,pattern=&quot;_indivresults&quot;)
names(table)&lt;-c(&quot;model&quot;,&quot;rg&quot;,&quot;SE&quot;,&quot;CI&quot;,&quot;p_value&quot;,&quot;p_adjust&quot;)

library(DT)
datatable(table, caption = &quot;Genetic correlations between individual networks and specific cognitive abilities&quot;, rownames=FALSE,options = list(pageLength=5, scrollX=T))</code></pre>
<div id="htmlwidget-8a36fcd57774400bb5d4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8a36fcd57774400bb5d4">{"x":{"filter":"none","caption":"<caption>Genetic correlations between individual networks and specific cognitive abilities<\/caption>","data":[["central_exec_Matrix","central_exec_Memory","central_exec_Symbol Digit","cingulo_Matrix","cingulo_Memory","cingulo_Symbol Digit","default_mode_Matrix","default_mode_Memory","default_mode_Symbol Digit","hippocampal_Matrix","hippocampal_Memory","hippocampal_Symbol Digit","multiple_Matrix","multiple_Memory","multiple_Symbol Digit","p_fit_Matrix","p_fit_Memory","p_fit_Symbol Digit","salience_Matrix","salience_Memory","salience_Symbol Digit","sensori_Matrix","sensori_Memory","sensori_Symbol Digit","temporo_Matrix","temporo_Memory","temporo_Symbol Digit","whole_brain_Matrix","whole_brain_Memory","whole_brain_Symbol Digit"],[0.2013,0.1168,0.1403,0.1907,0.0915,0.1209,0.2219,0.1082,0.1387,0.185,0.0278,0.1263,0.1675,0.111,0.11,0.227,0.1139,0.1204,0.1276,0.0498,0.1285,0.1111,0.0888,0.1198,0.2416,0.0967,0.1087,0.2287,0.0991,0.1365],[0.0884,0.0384,0.0463,0.095,0.0369,0.0458,0.0913,0.0405,0.0446,0.094,0.04,0.0452,0.0888,0.0393,0.0465,0.0865,0.0382,0.0456,0.0912,0.0354,0.047,0.0877,0.0401,0.0468,0.0881,0.039,0.0461,0.0875,0.0384,0.0469],["0.0281 - 0.3745","0.0415 - 0.1921","0.0495 - 0.2311","0.0046 - 0.3768","0.0192 - 0.1638","0.0311 - 0.2107","0.043 - 0.4008","0.0288 - 0.1876","0.0513 - 0.2261","7e-04 - 0.3693","-0.0505 - 0.1061","0.0376 - 0.2149","-0.0066 - 0.3416","0.0339 - 0.188","0.0189 - 0.2011","0.0575 - 0.3964","0.039 - 0.1887","0.031 - 0.2097","-0.0512 - 0.3065","-0.0197 - 0.1192","0.0363 - 0.2207","-0.0608 - 0.283","0.0103 - 0.1673","0.0282 - 0.2115","0.069 - 0.4143","0.0202 - 0.1732","0.0184 - 0.199","0.0571 - 0.4002","0.0238 - 0.1745","0.0446 - 0.2284"],[0.0227,0.0024,0.0024,0.0446,0.0131,0.0083,0.0151,0.0075,0.0019,0.0491,0.4866,0.0052,0.0593,0.0048,0.018,0.0087,0.0029,0.0083,0.1619,0.16,0.0063,0.2053,0.0267,0.0104,0.0061,0.0133,0.0182,0.009,0.0099,0.0036],[0.031,0.0193,0.0193,0.0558,0.0221,0.0193,0.0238,0.0193,0.0193,0.0589,0.4866,0.0193,0.0684,0.0193,0.0261,0.0193,0.0193,0.0193,0.1735,0.1735,0.0193,0.2124,0.0348,0.0195,0.0193,0.0221,0.0261,0.0193,0.0195,0.0193]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>model<\/th>\n      <th>rg<\/th>\n      <th>SE<\/th>\n      <th>CI<\/th>\n      <th>p_value<\/th>\n      <th>p_adjust<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2,4,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><strong>Most genetic associations between the networks and individual cognitive traits were significant.</strong></p>
<p><br></p>
<p><strong>The following table lists the traits that were non-significantly associated:</strong></p>
<pre class="r"><code>## which results were significant?
non_sig_models&lt;-table[which(table$p_adjust&gt;0.05),]


library(tidyr)
non_sig_models&lt;-separate(non_sig_models,col=&quot;model&quot;,into=c(&quot;network&quot;,&quot;cognitive_trait&quot;),sep = &quot;_&quot;)

kable(non_sig_models,row.names = F,caption = &quot;Genetic correlations that were non-significant at an FDR adjusted p-value larger than 5%&quot;,digits = 2)</code></pre>
<table>
<caption>Genetic correlations that were non-significant at an FDR adjusted p-value larger than 5%</caption>
<thead>
<tr class="header">
<th align="left">network</th>
<th align="left">cognitive_trait</th>
<th align="right">rg</th>
<th align="right">SE</th>
<th align="left">CI</th>
<th align="right">p_value</th>
<th align="right">p_adjust</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">cingulo</td>
<td align="left">Matrix</td>
<td align="right">0.19</td>
<td align="right">0.10</td>
<td align="left">0.0046 - 0.3768</td>
<td align="right">0.04</td>
<td align="right">0.06</td>
</tr>
<tr class="even">
<td align="left">hippocampal</td>
<td align="left">Matrix</td>
<td align="right">0.18</td>
<td align="right">0.09</td>
<td align="left">7e-04 - 0.3693</td>
<td align="right">0.05</td>
<td align="right">0.06</td>
</tr>
<tr class="odd">
<td align="left">hippocampal</td>
<td align="left">Memory</td>
<td align="right">0.03</td>
<td align="right">0.04</td>
<td align="left">-0.0505 - 0.1061</td>
<td align="right">0.49</td>
<td align="right">0.49</td>
</tr>
<tr class="even">
<td align="left">multiple</td>
<td align="left">Matrix</td>
<td align="right">0.17</td>
<td align="right">0.09</td>
<td align="left">-0.0066 - 0.3416</td>
<td align="right">0.06</td>
<td align="right">0.07</td>
</tr>
<tr class="odd">
<td align="left">salience</td>
<td align="left">Matrix</td>
<td align="right">0.13</td>
<td align="right">0.09</td>
<td align="left">-0.0512 - 0.3065</td>
<td align="right">0.16</td>
<td align="right">0.17</td>
</tr>
<tr class="even">
<td align="left">salience</td>
<td align="left">Memory</td>
<td align="right">0.05</td>
<td align="right">0.04</td>
<td align="left">-0.0197 - 0.1192</td>
<td align="right">0.16</td>
<td align="right">0.17</td>
</tr>
<tr class="odd">
<td align="left">sensori</td>
<td align="left">Matrix</td>
<td align="right">0.11</td>
<td align="right">0.09</td>
<td align="left">-0.0608 - 0.283</td>
<td align="right">0.21</td>
<td align="right">0.21</td>
</tr>
</tbody>
</table>
<p><br> <br></p>
</div>
<div id="qtrait" class="section level2">
<h2>Qtrait</h2>
<p>The analyses above, aiming to determine contributions of specific cognitive abilities to the genetic associations between general cognitive ability and networks, are somewhat inclusive. Most traits were significantly associated with the networks. We employ a technique called <a href="https://www.medrxiv.org/content/10.1101/2020.09.22.20196089v1.full.pdf">Qtrait heterogeneity index</a> to disentangle whether model fit significantly declines, if the model described above allows for direct associations between the networks and specific cognitive abilities. A significant Qtrait statistic would suggest that model fit significantly declined and that the association of interest is likely driven by specific cognitive abilities, rather than going through the general cognitive ability factor.</p>
<p>In the following, we fit two models as displayed in the figure below, through which we obtain chi-squared fit statistics for each model. The Qtrait heterogeneity index is calculated as the difference between the two fit statistics, and we test whether this difference significantly differs from zero. We repeat this procedure 10 times, once for each network. Qtraits statistics with p-values below 0.05/10 indicate that genetic associations between general cognitive ability and brain networks probably don’t go through the cognitive ability factor, but are driven by variation in specific cognitive traits.</p>
<p><img src="Path_diagram_Qtrait.jpg" /></p>
<pre class="r"><code>#load dependencies

library(stringr)
library(data.table)
#library(devtools)
library(GenomicSEM)

# load correlation matrix
temporarywd&lt;-paste0(workingd,&quot;/data_my_own/ldsc_networks/&quot;)
setwd(temporarywd)

load(&quot;all_networks_cognitive.RData&quot;)
ldscoutput&lt;-LDSCoutput

trait_names&lt;-dimnames(ldscoutput$S)[[2]]

dimnames(ldscoutput$S)[[1]]&lt;-trait_names
dimnames(ldscoutput$S)[[2]]&lt;-trait_names

# list all networks names
network_names&lt;-c(&quot;central_exec&quot;,&quot;cingulo&quot;,&quot;default_mode&quot;,&quot;hippocampal&quot;,&quot;multiple&quot;,&quot;p_fit&quot;,&quot;salience&quot;,&quot;sensori&quot;,&quot;temporo&quot;,&quot;whole_brain&quot;)


# build dataframe to hold the Qtrait results
table &lt;- as.data.frame(matrix(nrow = 10, ncol = 4))
names(table)&lt;-c(&quot;network&quot;,&quot;Qtrait&quot;,&quot;df&quot;,&quot;pval&quot;)

#### genomic SEM

for (i in network_names){
            
            # specify covariance structure of interest
            covstruc &lt;- ldscoutput

            #specify estimation method
            estimation&lt;- &quot;DWLS&quot;

            ### specify your model that assumes all associations go through genetic g factor
            #PC_model&lt;- &#39;G =~ NA*Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR
            #           G ~~ 1*G 
            #           G ~ central_exec&#39;
            
            factor_model&lt;-paste0(&quot;G =~ Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR\n\t\t\t\t\t\tG ~ &quot;,network_name)

            # run your model    
            factor_model&lt;-usermodel(covstruc=covstruc, model=factor_model, estimation=estimation, imp_cov=T)
            
            
            ### specify your model that allows specific cognitive traits to be directly associated with network
            indep_model&lt;- paste0(&quot;G =~ Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR
                      &quot;,i,&quot; ~ Matrix+Memory+RT+Symbol Digit+TMTB+Tower+VNR&quot;)
            # run model    
            indep_model&lt;-usermodel(covstruc=covstruc, model=indep_model, estimation=estimation, imp_cov=T)
            
            diff_chisq &lt;- as.numeric(factor_model$modelfit[3] - indep_model$modelfit[3])
            diff_df &lt;- as.numeric(factor_model$modelfit[2] - indep_model$modelfit[2])
            qtrait_pval &lt;- 1-pchisq(diff_chisq, diff_df)
            
            # save qtrait results in table
            table$network[which(network_names == i)] &lt;- i
            table$Qtrait[which(network_names == i)] &lt;- diff_chisq
            table$df[which(network_names == i)] &lt;- diff_df
            table$pval[which(network_names == i)] &lt;- qtrait_pval
}

table$Qtrait &lt;- signif(table$Qtrait, digits = 2)</code></pre>
<pre class="r"><code>kable(table,row.names = FALSE,caption = &quot;Qtrait statistics&quot;, digits = 24)</code></pre>
<table>
<caption>Qtrait statistics</caption>
<thead>
<tr class="header">
<th align="left">network</th>
<th align="right">Qtrait</th>
<th align="right">df</th>
<th align="right">pval</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">central_exec</td>
<td align="right">-2.1e-19</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">cingulo</td>
<td align="right">-4.7e-20</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">default_mode</td>
<td align="right">-1.8e-19</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">hippocampal</td>
<td align="right">-1.5e-19</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">multiple</td>
<td align="right">-4.9e-20</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">p_fit</td>
<td align="right">-5.9e-20</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">salience</td>
<td align="right">-4.5e-19</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">sensori</td>
<td align="right">-6.0e-20</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">temporo</td>
<td align="right">-5.5e-20</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">whole_brain</td>
<td align="right">-1.0e-19</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p>This indicates that genetic correlations between general cognitive ability and brain networks are unlikely driven by specific cognitive traits. Allowing the specific traits to directly correlate with a brain network (i.e. not where the correlation goes through the factor) did not increase the model fit. It seems that these associations are not limited to a single measures and are much more general.</p>
<hr />
</div>
</div>
<div id="genetic-correlation-with-brain-age" class="section level1">
<h1>Genetic correlation with brain age</h1>
<p>Previous research modelling phenotypic canonical networks showed that larger shared morphology within the whole brain (PC loadings) were negatively associated with age sensitivity, even when connectivity measures had been residualised for age [34]. Consequently, the authors suggested that previously described patterns of brain structural aging, a construct labelled as brain age <a href="https://www.sciencedirect.com/science/article/pii/S0006322320316796">Ref</a>, might not capture how quickly ones regional volumes decline compared to their peers, but rather healthy morphology across the brain. This could vary between people for many non-age-related reasons, including genetic predisposition. Genetic predisposition towards strong whole-brain connectivity might contribute to healthier well-connected brains, that could be more resilient towards harmful environmental factors, properties that should be captured by younger brain age. To test this theory, we calculated genetic correlations between shared morphology across the whole-brain and brain age. Bivariate LDSC was calculated between whole-brain PC summary statistics, created using the methods demonstrated above, and brain age GWAS summary statistics <a href="https://www.nature.com/articles/s41593-019-0471-7">REF</a>.</p>
<pre class="r"><code>## this script calculates the genetic correlation between whole brain PC sumstats and brainage gwas 
# brain age gwas was downloaded from https://github.com/tobias-kaufmann/brainage/find/master
# fileparts combined using: cat Brainage_GWAS_sumstat_final.gz* &gt; ~/data/brain_age_GWAS/Brainage_GWAS_sumstat_final.gz
# unzipped using gunzip Brainage_GWAS_sumstat_final.gz

# load dependencies
#library(devtools)
library(GenomicSEM)
library(data.table)

#########################
## format brain age file
########################
setwd(&quot;~/data/brain_age_GWAS/&quot;)
brainage&lt;-fread(&quot;Brainage_GWAS_sumstat_final.txt&quot;,header=T,data.table=F)

brainage$N&lt;-45615

fwrite(brainage, file=&quot;~/data/brain_age_GWAS/brain_age_gwas_header.txt&quot;,quote=FALSE,col.names=TRUE,row.names=F,sep=&quot; &quot;,na=&quot;NA&quot;)


###################
## munge brainage gwas

hm3&lt;-&quot;~/data/w_hm3.noMHC.snplist&quot;

files&lt;-&quot;brain_age_gwas_header.txt&quot;
trait.names&lt;-&quot;brainage&quot;

munge(files=files,
        hm3=hm3,
        trait.names=trait.names,
        info.filter = 0.9)

################################################
### LDSC brain age and whole brain PC 
################################################

# specify traits and their location
traits&lt;-c(&quot;~/output/genetic_networks_project/brainage/brainage.sumstats.gz&quot;,&quot;~/output/genetic_networks_project/PC_munged/whole_brain.sumstats.gz&quot;)
trait.names&lt;-c(&quot;brainage&quot;,&quot;whole_brain_PC&quot;)


# ld scores and weights previously downloaded (European population)
ld&lt;-&quot;~/data/eur_w_ld_chr/&quot;
wld&lt;-&quot;~/data/eur_w_ld_chr/&quot;


# no sample or population prevalence required because they are continuous traits
desired_length&lt;-length(traits)
sample.prev&lt;-rep(NA, desired_length)
population.prev&lt;-rep(NA, desired_length)

LDSCoutput&lt;-ldsc(traits=traits,
                    ld=ld,wld=wld,
                    trait.names = trait.names,
                    ldsc.log=&quot;~/output/genetic_networks_project/brainage/whole_brain_vs_brainage.log&quot;,
                    sample.prev=sample.prev,
                    population.prev=population.prev,
                    stand=T)
                    
save(LDSCoutput, file=&quot;~/output/genetic_networks_project/brainage/whole_brain_vs_brainage.RData&quot;)</code></pre>
<pre class="r"><code>setwd(paste0(workingd,&quot;/data_my_own/brainage&quot;))
load(&quot;whole_brain_vs_brainage.RData&quot;)
brainage_cor&lt;-round(LDSCoutput$S_Stand[1,2], digits = 2)
k&lt;-nrow(LDSCoutput$S)
SE&lt;-matrix(0, k, k)
SE[lower.tri(SE,diag=TRUE)]&lt;-sqrt(diag(LDSCoutput$V_Stand))
SE_brainage&lt;-round(SE[2,1],digits=3)</code></pre>
<p>We found that younger brains were genetically correlated with brains demonstrating higher shared morphology (<strong>rg = -0.34; SE = 0.065</strong>), indicating that the neurostructural patterns shared by both concepts on a phenotypic level are at least partly governed by polygenic effects of brain health, rather than biological age-specific processes per se.</p>
</div>

&nbsp;
<hr />
<a href = "https://www.kcl.ac.uk/">
<p style="text-align: center"><img src="KCL_logo.jpg" href="https://www.kcl.ac.uk/" style="width:70px;height:50px"> 
</a>
</p>

<a href = "https://www.kcl.ac.uk/people/anna-furtjes">
<p style="text-align: center;">By Anna Elisabeth Fürtjes</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>anna.furtjes@kcl.ac.uk</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
    <a href="https://twitter.com/Anna_Furtjes" class="fa fa-twitter"></a>
</p>

&nbsp;


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
